---
title: "TB_LION_RNAseq_analysis"
author: "avanvalken"
date: "2024-10-02"
output: html_document
---
```{r setup, include=FALSE}

library(tidyverse)
library("readxl")
library(SummarizedExperiment)
library(SingleCellExperiment)
library(singleCellTK)
library(cowplot)
library(patchwork)
library(TBSignatureProfiler)
library(umap)
library(limma)
library(ComplexHeatmap)
library(DT)

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(results='hide')


```

# Load Data {.tabset}
```{r}
#se <- readRDS("data/tblion_se.RDS")
se <- readRDS("data/tb_lion_rsubread_se.RDS")

df <- as.data.frame(colData(se))
```

## change nutrition status to 16<="mal"
```{r}
# colData(se)$nutrition <- ifelse(colData(se)$BMI > 16, "well", "mal")
# df <- as.data.frame(colData(se))
```


# Explore the data {.tabset}

## visit number, nutrition status, helminth status, male/female, AGE {.tabset}
```{r}

# Plot visit, color by nutrition status, then helminth status
p1 <- ggplot(data=df, mapping=aes(x=visit, y=AGE, fill=nutrition)) +
  geom_boxplot()


p2 <- ggplot(data=df, mapping=aes(x=visit, y=AGE, fill=GENDER)) +
  geom_boxplot()

p3 <- ggplot(data=df, mapping=aes(x=visit, y=BMI, fill=HELMINTH)) +
  geom_boxplot()

p4 <- ggplot(data=df, mapping=aes(x=visit, y=BMI, fill=nutrition)) +
  geom_boxplot()

p5 <- ggplot(data=df, mapping=aes(x=visit, y=BMI, fill=GENDER)) +
  geom_boxplot()


p1 + p2 

p3 + p4

p5 
```

## visit number, nutrition status, helminth status, male/female, AGE {.tabset}
```{r}

# Plot visit, color by nutrition status, then helminth status
p1 <- ggplot(data=df, mapping=aes(x=visit, y=AGE, fill=status)) +
  geom_boxplot()




p2 <- ggplot(data=df, mapping=aes(x=visit, y=BMI, fill=status)) +
  geom_boxplot()




p1 / p2 


```
# V1 Demographics
```{r}
df <- as.data.frame(colData(se))
v1 <- filter(df, visit=="Visit_1")

table(v1$HELMINTH, v1$nutrition)
```


# Analyze SE object for preliminary grant data {.tabset}
```{r}
# counts to cpm/log/log_counts_cpm

# Want to have 5% present rate
se <- se[apply(assay(se,"counts") != 0, 1, mean)>.2,] 

# add assays (log2, counts_per_million, log_counts_cpm)
se <- mkAssay(se, log=T)
names(assays(se))

indata <- se
```
## subset by visit 1 and visit 6 {.tabset}
```{r}
# should V7 be the same as V6?
indata_v1 <- indata[,colData(indata)$visit=="Visit_1"] 
indata_v1@colData@listData[["visit"]]

indata_v6 <- indata[,colData(indata)$visit=="Visit_6"]
indata_v6@colData@listData[["visit"]]
```

# Dimension Reduction all samples  {.tabset}


## PCA  {.tabset}
Using the DESeq object and PCAplot function on the ComBat corrected data



```{r}
# path to save
path <- file.path("outs/dimension_reduction", "all_samples")
#dir.create(path)


indata <- se


# helminth/nutrition info
table(colData(indata)$"status")


indata_tmp <- SummarizedExperiment(assays=list(counts=assays(indata)$log_counts_cpm), colData = colData(indata))

#indata_tmp <- indata_tmp[,-which(colnames(indata_tmp) %in% c("10200107A0"))]

names(indata_tmp) <- NULL

indata_tmp2 <- as(indata_tmp, "SingleCellExperiment")
#plotPCA(DESeqTransform(indata_tmp))
# g1 <- DESeq2::plotPCA(DESeq2::DESeqTransform(indata_tmp), intgroup="status")
# g1

g <- singleCellTK::plotPCA(indata_tmp2, colorBy = "status", shape="visit", useAssay = "counts", runPCA = TRUE)
g <- g + geom_point(size = 3)
g


```
### Save
```{r}

ggsave(file.path(path, "pca_sctk_all.svg"),plot=g, width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "pca_sctk_all.png"),plot=g, width = 3.5, height = 3.5, pointsize = 5)

```


## UMAP {.tabset}

```{r }
assay_type = "log_counts_cpm"

set.seed(1)
umap_out <- umap(t(assay(indata,assay_type)))

embedding <- as.data.frame(umap_out$layout)
embedding$Class <- as.factor(indata$nutrition)
embedding$Helminth <- as.factor(indata$HELMINTH)
embedding$status <- as.factor(indata$status)
embedding$visit <- as.factor(indata$visit)

g <- ggplot(embedding, aes(x=V1, y=V2, color=status, shape=visit, label = colnames(assay(indata,assay_type)))) + geom_point(size=3) + xlab("UMAP 1") + 
  ylab("UMAP 2") + 
  theme(plot.title = element_text(hjust = 0.5)) 

plot(g)

g
```
### Save
```{r}
ggsave(file.path(path, "umap_all.svg"), width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "umap_all.png"), width = 3.5, height = 3.5, pointsize = 5)

```

## Mal-only, V1 and V6
```{r}
indata.mal <- se[,colData(indata)$nutrition=="mal"]
indata <- indata.mal
```

#### PCA  {.tabset}
Using the DESeq object and PCAplot function on the ComBat corrected data

```{r}
## path to save
path <- file.path("outs/dimension_reduction", "all_samples")
##dir.create(path)


## helminth/nutrition info
table(colData(indata)$"status")


indata_tmp <- SummarizedExperiment(assays=list(counts=assays(indata)$log_counts_cpm), colData = colData(indata))

##indata_tmp <- indata_tmp[,-which(colnames(indata_tmp) %in% c("10200107A0"))]

names(indata_tmp) <- NULL

indata_tmp2 <- as(indata_tmp, "SingleCellExperiment")
##plotPCA(DESeqTransform(indata_tmp))
## g1 <- DESeq2::plotPCA(DESeq2::DESeqTransform(indata_tmp), intgroup="status")
## g1

g <- singleCellTK::plotPCA(indata_tmp2, colorBy = "status", shape="visit", useAssay = "counts", runPCA = TRUE)
g <- g + geom_point(size = 3)
g


```
###### Save
```{r}

ggsave(file.path(path, "pca_sctk_mal.svg"),plot=g, width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "pca_sctk_mal.png"),plot=g, width = 3.5, height = 3.5, pointsize = 5)

```


#### UMAP {.tabset}

```{r }
assay_type = "log_counts_cpm"

set.seed(1)
umap_out <- umap(t(assay(indata,assay_type)))

embedding <- as.data.frame(umap_out$layout)
embedding$Class <- as.factor(indata$nutrition)
embedding$Helminth <- as.factor(indata$HELMINTH)
embedding$status <- as.factor(indata$status)
embedding$visit <- as.factor(indata$visit)

g <- ggplot(embedding, aes(x=V1, y=V2, color=status, shape=visit, label = colnames(assay(indata,assay_type)))) + geom_point(size=3) + xlab("UMAP 1") + 
  ylab("UMAP 2") + 
  theme(plot.title = element_text(hjust = 0.5)) 

plot(g)

g
```
###### Save
```{r}
ggsave(file.path(path, "umap_mal.svg"), width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "umap_mal.png"), width = 3.5, height = 3.5, pointsize = 5)

```


## Absent_mal, V1 and V6
```{r}
indata <- se[,colData(se)$status=="absent_mal"]

```

#### PCA  {.tabset}
Using the DESeq object and PCAplot function on the ComBat corrected data

```{r}
## path to save
path <- file.path("outs/dimension_reduction", "all_samples")
##dir.create(path)


## helminth/nutrition info
table(colData(indata)$"status")


indata_tmp <- SummarizedExperiment(assays=list(counts=assays(indata)$log_counts_cpm), colData = colData(indata))

##indata_tmp <- indata_tmp[,-which(colnames(indata_tmp) %in% c("10200107A0"))]

names(indata_tmp) <- NULL

indata_tmp2 <- as(indata_tmp, "SingleCellExperiment")
##plotPCA(DESeqTransform(indata_tmp))
## g1 <- DESeq2::plotPCA(DESeq2::DESeqTransform(indata_tmp), intgroup="status")
## g1

g <- singleCellTK::plotPCA(indata_tmp2, colorBy = "visit", useAssay = "counts", runPCA = TRUE)
g <- g + geom_point(size = 3)
g


```
###### Save
```{r}

ggsave(file.path(path, "pca_sctk_absent_mal.svg"),plot=g, width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "pca_sctk_absent_mal.png"),plot=g, width = 3.5, height = 3.5, pointsize = 5)

```


#### UMAP {.tabset}

```{r }
assay_type = "log_counts_cpm"

set.seed(1)
umap_out <- umap(t(assay(indata,assay_type)))

embedding <- as.data.frame(umap_out$layout)
embedding$Class <- as.factor(indata$nutrition)
embedding$Helminth <- as.factor(indata$HELMINTH)
embedding$status <- as.factor(indata$status)
embedding$visit <- as.factor(indata$visit)

g <- ggplot(embedding, aes(x=V1, y=V2, color=visit, label = colnames(assay(indata,assay_type)))) + geom_point(size=3) + xlab("UMAP 1") + 
  ylab("UMAP 2") + 
  theme(plot.title = element_text(hjust = 0.5)) 

plot(g)

g
```
###### Save
```{r}
ggsave(file.path(path, "umap_mal_absent_mal.svg"), width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "umap_mal_absent_mal.png"), width = 3.5, height = 3.5, pointsize = 5)

```


## present_mal, V1 and V6
```{r}
indata <- se[,colData(se)$status=="present_mal"]

```

#### PCA  {.tabset}
Using the DESeq object and PCAplot function on the ComBat corrected data

```{r}
## path to save
path <- file.path("outs/dimension_reduction", "all_samples")
##dir.create(path)


## helminth/nutrition info
table(colData(indata)$"status")


indata_tmp <- SummarizedExperiment(assays=list(counts=assays(indata)$log_counts_cpm), colData = colData(indata))

##indata_tmp <- indata_tmp[,-which(colnames(indata_tmp) %in% c("10200107A0"))]

names(indata_tmp) <- NULL

indata_tmp2 <- as(indata_tmp, "SingleCellExperiment")
##plotPCA(DESeqTransform(indata_tmp))
## g1 <- DESeq2::plotPCA(DESeq2::DESeqTransform(indata_tmp), intgroup="status")
## g1

g <- singleCellTK::plotPCA(indata_tmp2, colorBy = "visit", useAssay = "counts", runPCA = TRUE)
g <- g + geom_point(size = 3)
g


```
###### Save
```{r}

ggsave(file.path(path, "pca_sctk_present_mal.svg"),plot=g, width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "pca_sctk_present_mal.png"),plot=g, width = 3.5, height = 3.5, pointsize = 5)

```


#### UMAP {.tabset}

```{r }
assay_type = "log_counts_cpm"

set.seed(1)
umap_out <- umap(t(assay(indata,assay_type)))

embedding <- as.data.frame(umap_out$layout)
embedding$Class <- as.factor(indata$nutrition)
embedding$Helminth <- as.factor(indata$HELMINTH)
embedding$status <- as.factor(indata$status)
embedding$visit <- as.factor(indata$visit)

g <- ggplot(embedding, aes(x=V1, y=V2, color=visit, label = colnames(assay(indata,assay_type)))) + geom_point(size=3) + xlab("UMAP 1") + 
  ylab("UMAP 2") + 
  theme(plot.title = element_text(hjust = 0.5)) 

plot(g)

g
```
###### Save
```{r}
ggsave(file.path(path, "umap_mal_present_mal.svg"), width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "umap_mal_present_mal.png"), width = 3.5, height = 3.5, pointsize = 5)

```




# Dimension Reduction Visit 1  {.tabset}


## PCA  {.tabset}
Using the DESeq object and PCAplot function on the ComBat corrected data



```{r}
# path to save
path <- file.path("outs/dimension_reduction", "visit_1")
#dir.create(path)
#path1 <- file.path(path, "visit_1")
#dir.create(path1)
# filter by time
indata <- indata_v1


# helminth/nutrition info
table(colData(indata)$"status")


indata_tmp <- SummarizedExperiment(assays=list(counts=assays(indata)$log_counts_cpm), colData = colData(indata))

#indata_tmp <- indata_tmp[,-which(colnames(indata_tmp) %in% c("10200107A0"))]

names(indata_tmp) <- NULL

indata_tmp2 <- as(indata_tmp, "SingleCellExperiment")
#plotPCA(DESeqTransform(indata_tmp))
g1 <- DESeq2::plotPCA(DESeq2::DESeqTransform(indata_tmp), intgroup="status")
g1

g <- singleCellTK::plotPCA(indata_tmp2, colorBy = "nutrition",shape="HELMINTH", useAssay = "counts", runPCA = TRUE)
g <- g + geom_point(size = 3)
g

g1 + g
```
### Save
```{r}
ggsave(file.path(path, "pca_deseq2.svg"),plot=g1, width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "pca_sctk.svg"),plot=g, width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "pca_deseq2.png"),plot=g1, width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "pca_sctk.png"),plot=g, width = 3.5, height = 3.5, pointsize = 5)

```


## UMAP {.tabset}

```{r }
assay_type = "log_counts_cpm"

set.seed(1)
umap_out <- umap(t(assay(indata,assay_type)))

embedding <- as.data.frame(umap_out$layout)
embedding$Class <- as.factor(indata$nutrition)
embedding$Helminth <- as.factor(indata$HELMINTH)
embedding$status <- as.factor(indata$status)

g <- ggplot(embedding, aes(x=V1, y=V2, color=Class, shape=Helminth, label = colnames(assay(indata,assay_type)))) + geom_point(size=3) + xlab("UMAP 1") + 
  ylab("UMAP 2") + 
  theme(plot.title = element_text(hjust = 0.5)) 

plot(g)

g
```
### Save
```{r}
ggsave(file.path(path, "umap.svg"), width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "umap.png"), width = 3.5, height = 3.5, pointsize = 5)

```


# Differential Expression Visit 1 {.tabset}
Create 4 heatmaps of filtered data. 2 heatmaps of data with confounding factors acounted for. One heatmap of well-nourished helminth absent vs all and see differences, determine if groups cluster together. Get pathway gene list of which makes the most sense.
```{r}
# data subset
indata <- se[,colData(se)$visit=="Visit_1"]

# save for IPA
ipa <- file.path("outs/ipa/ipa_upload")
```


## Helminth vs no Helminth, control for nutrition {.tabset}
```{r }
visit <- "v1_"
diffex_test <- "helPres_v_helAbs_con_nutrition_v1_"
path <- file.path("outs/differential_expression/visit_1/helPres_v_helAbs_con_nutrition")
dir.create(path, recursive = T)
# matrix for differential expression
designMat <- model.matrix(~factor(HELMINTH) + factor(nutrition), data=colData(indata))

colnames(designMat)
colnames(designMat) <- c("Intercept", "Present", "well")

head(designMat)

fit <- lmFit(assay(indata, "log_counts_cpm"), designMat)

# Difference in expression between Failures and controls 

contrast.matrixNut<- makeContrasts(Present, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata)$subjtype)

new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
##shows spike near 0

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]


# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]

```



### 2 All genes datatable and heatmap {.tabset}
```{r }
DT::datatable(sig_limmaResNut, 
          options=list(scrollX=T,pAGELength=20),
          rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid") %>% group_by(Helminth) %>% arrange(Helminth, BMI)
mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(Helminth, Nutrition))

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))

### plot heatmap
g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )
g

g1 <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = F )
g + g1

```

### Save
```{r}
# genes
write.csv(sig_limmaResNut, file.path(path, paste0(visit,"helminth_con_nutrition_sig.csv")))
write.csv(sig_limmaResNut, file.path(ipa, paste0(visit,"helminth_con_nutrition_sig.csv")))

# save plot
svg(file.path(path, "helminth_con_nutrition_sig.svg"), width=7, height=7)
g
dev.off()
png(file.path(path, "helminth_con_nutrition_sig.png"), width=700, height=700)
g
dev.off()

# save plot
svg(file.path(path, "helminth_con_nutrition_sig_ordered.svg"), width=7, height=7)
g1
dev.off()
png(file.path(path, "helminth_con_nutrition_sig_ordered.png"), width=700, height=700)
g1
dev.off()
```



number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

## Helminth vs no Helminth, mal-nourished only {.tabset}
```{r }
path <- file.path("outs/differential_expression/visit_1/mal_only_helPos_v_helAbs")
dir.create(path)
indata.mal <- indata[,colData(indata)$nutrition=="mal"]
# matrix for differential expression
designMat <- model.matrix(~factor(HELMINTH) , data=colData(indata.mal)) # with mal < 16; no Helminth negative

colnames(designMat)
colnames(designMat) <- c("Intercept", "Present")

head(designMat)

fit <- lmFit(assay(indata.mal, "log_counts_cpm"), designMat)

# Difference in expression between Failures and controls 

contrast.matrixNut<- makeContrasts(Present, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata)$subjtype)

new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)

new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)

##shows spike near 0

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]


# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]


```



### 2 All genes datatable and heatmap {.tabset}
```{r }
DT::datatable(sig_limmaResNut, 
          options=list(scrollX=T,pAGELength=20),
          rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH") 


ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))
g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )

#write.csv(sig_limmaResNut, file.path(path, "helminth_mal_only_sig.csv"))
write.csv(sig_limmaResNut, file.path(path, paste0(visit,"helminth_con_nutrition_sig.csv")))
# save plot
svg(file.path(path, "helminth_mal_only_sig.svg"), width=7, height=7)
g
dev.off()
png(file.path(path, "helminth_mal_only_sig.png"), width=700, height=700)
g
dev.off()

g
```

number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

## Helminth vs no Helminth, well-nourished only {.tabset}
```{r }
path <- file.path("outs/differential_expression", "visit_1", "well_only_helPos_v_helAbs")
dir.create(path)
indata.well <- indata[,colData(indata)$nutrition=="well"]
# matrix for differential expression
designMat <- model.matrix(~factor(HELMINTH) , data=colData(indata.well))

colnames(designMat)
colnames(designMat) <- c("Intercept", "Present")

head(designMat)

fit <- lmFit(assay(indata.well, "log_counts_cpm"), designMat)

# Difference in expression between Failures and controls 

contrast.matrixNut<- makeContrasts(Present, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata)$subjtype)

new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
#new_hist_padjval_100breaks
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
#new_hist_pval_100breaks
##shows spike near 0

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]


# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]


```



### 2 All genes datatable and heatmap {.tabset}
```{r }
DT::datatable(sig_limmaResNut, 
          options=list(scrollX=T,pAGELength=20),
          rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata.well[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
df=data.frame(Helminth=colData(indata.well)$"HELMINTH",
              Nutrition=colData(indata.well)$"nutrition") 


ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))
g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )
g
write.csv(sig_limmaResNut, file.path(path, paste0(visit,"helminth_abs_vs_pres_well-only_sig.csv")))

# save plot
svg(file.path(path, "helminth_abs_vs_pres_well-only_sig.svg"), width=7, height=7)
g
dev.off()
png(file.path(path, "helminth_abs_vs_pres_well-only_sig.png"), width=700, height=700)
g
dev.off()

g
```

number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

## mal vs well, control for helminth {.tabset}
### all significant genes
```{r }
path <- file.path("outs/differential_expression/visit_1","mal_vs_well_con_helminth")
dir.create(path)
# relevel so well is the intercept
colData(indata)$nutrition <- factor(colData(indata)$nutrition, levels=c("well","mal"))
levels(colData(indata)$nutrition)
# matrix for differential expression
designMat <- model.matrix(~nutrition + factor(HELMINTH) , data=colData(indata))

colnames(designMat)
colnames(designMat) <- c("Intercept", "mal", "Present")

head(designMat)

fit <- lmFit(assay(indata, "log_counts_cpm"), designMat)

# Difference in expression between well nourished and control for helminth status 

contrast.matrixNut<- makeContrasts(mal, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata)$nutrition)

# histograms of p values
new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
#new_hist_padjval_100breaks
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
#new_hist_pval_100breaks

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]


# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]

# top 25 high and low genes by logFC 
top_25_logfc <- sig_limmaResNut[order(sig_limmaResNut$logFC, decreasing=T),]
top_25_logfc <- top_25_logfc[1:25,]
top_25_neg_logfc <- sig_limmaResNut[order(sig_limmaResNut$logFC, decreasing = F),]
top_25_neg_logfc <- top_25_neg_logfc[1:25,]
```

### Volcano Plot
```{r}
library(ggrepel)
mat = as.matrix(assay(indata,"log_counts_cpm"))

# dataframe to read for volcano plot
df <- limmaResNut %>% rownames_to_column(var="gene_symbol")

# label genes
label_genes <- c(rownames(top_25_logfc), rownames(top_25_neg_logfc))
# Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2fc respectively positive or negative)
df$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP"
df$diffexpressed[df$logFC > 0.6 & df$adj.P.Val < 0.05] <- "UP"
# if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"<br /><br /><br />
df$diffexpressed[df$logFC < -0.6 & df$adj.P.Val < 0.05] <- "DOWN"
unique(df$diffexpressed)
df <- df %>% mutate(delabel=ifelse(gene_symbol %in% label_genes, df$gene_symbol, NA))



unique()# Create a basic volcano plot
ggplot(data = df, aes(x = logFC, y = -log10(adj.P.Val), col = diffexpressed,  label = delabel)) +
  geom_vline(xintercept = c(-0.6, 0.6), col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') +
               scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), # to set the colours of our variable<br /><br /><br />
                     labels = c("Downregulated", "Not significant", "Upregulated")) +
  geom_point(size=2) +
               labs(color = 'Expression in Malnourished', #legend_title, 
       x = expression("log"[2]*"FC"), y = expression("-log"[10]*"(adjusted p-value)")) +
               ggtitle('Gene Expression at Baseline') +
               geom_text_repel()
               

```

#### Save
```{r}
ggsave(file.path(path, "volcano.png"), width=7, height=7, units="in", dpi=300)
ggsave(file.path(path, "volcano.svg"), width=7, height=7, units="in", dpi=300)

```


### all significant genes datatable and heatmap {.tabset}
```{r }
datatable(sig_limmaResNut, 
          options=list(scrollX=T,pAGELength=20),
          rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
#df=data.frame(Nutrition=colData(indata)$"nutrition", Helminths=colData(indata)$"HELMINTH") 
### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid",
              status=colData(indata)$"status") %>% arrange( status)
mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(Nutrition,Helminth))

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))

### plot heatmap

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))

g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )

g1 <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = F )

g + g1
```

### save
```{r}
# save significant gene list
write.csv(sig_limmaResNut, file.path(path, paste0(visit,"mal_vs_well_con_helminth_sig.csv")))
write.csv(sig_limmaResNut, file.path(ipa, paste0(visit,"mal_vs_well_con_helminth_sig.csv")))


# save plot
svg(file.path(path, "mal_vs_well_con_helminth_sig.svg"), width=7, height=7)
g
dev.off()

png(file.path(path, "mal_vs_well_con_helminth_sig.png"), width=700, height=700)
g
dev.off()


# save plot
svg(file.path(path, "mal_vs_well_con_helminth_sig_ordered.svg"), width=7, height=7)
g1
dev.off()

png(file.path(path, "mal_vs_well_con_helminth_sig_ordered.png"), width=700, height=700)
g1
dev.off()
```

### top and bottom 25 by logFC
```{r}
## Make a Heatmap of top25 genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(top_25_logfc),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
#df=data.frame(Nutrition=colData(indata)$"nutrition", Helminths=colData(indata)$"HELMINTH") 
### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid",
              status=colData(indata)$"status") %>% arrange( BMI)
mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(Nutrition))

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue")))

### plot heatmap

#ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))


top25.bl <- Heatmap(mat,show_row_names=T,show_column_names = F, top_annotation = ha, cluster_columns = F )

mat = as.matrix(assay(indata[row.names(top_25_neg_logfc),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
#df=data.frame(Nutrition=colData(indata)$"nutrition", Helminths=colData(indata)$"HELMINTH") 
### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid",
              status=colData(indata)$"status") %>% arrange( BMI)
mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(Nutrition))

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue")))

neg25.bl <- Heatmap(mat,show_row_names=T,show_column_names = F, top_annotation = ha, cluster_columns = F )



top25.bl + neg25.bl
```

#### save
```{r}
# save top and bottom 25 by logfc in mal
write.csv(top_25_logfc, file.path(path, paste0(visit,"mal_vs_well_con_helminth_top25logfc_in_mal.csv")))
write.csv(top_25_neg_logfc, file.path(path, paste0(visit,"mal_vs_well_con_helminth_lowest25logfc_in_mal.csv")))

```





number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

## mal vs well, helminth positive{.tabset}
```{r }
path <- file.path("outs/differential_expression", "visit_1", "mal_v_well_helPos_only")
dir.create(path)
# select only helminth positive
indata.pos <- indata[,colData(indata)$HELMINTH=="Present"]
# matrix for differential expression
designMat <- model.matrix(~factor(nutrition), data=colData(indata.pos))

colnames(designMat)
colnames(designMat) <- c("Intercept", "well")

head(designMat)

fit <- lmFit(assay(indata.pos, "log_counts_cpm"), designMat)

# Difference in expression between well nourished and control for helminth status 

contrast.matrixNut<- makeContrasts(well, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata.pos)$nutrition)

# histograms of p values
new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
#new_hist_padjval_100breaks
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
#new_hist_pval_100breaks

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]


# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]

```



### 2 All genes datatable and heatmap {.tabset}
```{r }
datatable(sig_limmaResNut, 
          options=list(scrollX=T,pAGELength=20),
          rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata.pos[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
df=data.frame(Nutrition=colData(indata.pos)$"nutrition", Helminths=colData(indata.pos)$"HELMINTH") 


ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))

g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )

g

```

number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

### save
```{r}
write.csv(sig_limmaResNut, file.path(path, paste0(visit,"mal_v_well_helminthPos_sig.csv")))
# save plot
svg(file.path(path, "mal_v_well_helminthPos_sig.svg"), width=7, height=7)
g
dev.off()
png(file.path(path, "mal_v_well_helminthPos_sig.png"), width=700, height=700)
g
dev.off()
```




## mal vs well, helminth negative {.tabset}
```{r }
path <- file.path("outs/differential_expression","visit_1", "mal_v_well_helAbs_only")
dir.create(path)
indata.neg <- indata[,colData(indata)$HELMINTH=="Absent"]
# matrix for differential expression
designMat <- model.matrix(~factor(nutrition), data=colData(indata.neg))

colnames(designMat)
colnames(designMat) <- c("Intercept", "well")

head(designMat)

fit <- lmFit(assay(indata.neg, "log_counts_cpm"), designMat)

# Difference in expression between well nourished and control for helminth status 

contrast.matrixNut<- makeContrasts(well, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata.neg)$nutrition)

# histograms of p values
new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
#new_hist_padjval_100breaks
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
#new_hist_pval_100breaks

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]

# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]

```



### 2 All genes datatable and heatmap {.tabset}
```{r }
datatable(sig_limmaResNut, 
          options=list(scrollX=T,pAGELength=20),
          rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata.neg[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
df=data.frame(Nutrition=colData(indata.neg)$"nutrition", Helminths=colData(indata.neg)$"HELMINTH") 


ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))

g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )

# save plot
write.csv(sig_limmaResNut, file.path(path,paste0(visit,"mal_v_well_helminthNeg_sig.csv")))
# save plot
svg(file.path(path, "mal_v_well_helminthNeg_sig.svg"), width=7, height=7)
g
dev.off()
png(file.path(path, "mal_v_well_helminthNeg_sig.png"), width=700, height=700)
g
dev.off()
g
```

number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

## Visit 6

# Dimension Reduction Visit 6  {.tabset}


## PCA  {.tabset}
Using the DESeq object and PCAplot function on the ComBat corrected data



```{r}
# path to save
path <- file.path("outs/dimension_reduction", "visit_6")
#dir.create(path)

#dir.create(path1)
# filter by time
indata <- indata_v6


# helminth/nutrition info
table(colData(indata)$"status")


indata_tmp <- SummarizedExperiment(assays=list(counts=assays(indata)$log_counts_cpm), colData = colData(indata))

#indata_tmp <- indata_tmp[,-which(colnames(indata_tmp) %in% c("10200107A0"))]

names(indata_tmp) <- NULL

indata_tmp2 <- as(indata_tmp, "SingleCellExperiment")
#plotPCA(DESeqTransform(indata_tmp))
g1 <- DESeq2::plotPCA(DESeq2::DESeqTransform(indata_tmp), intgroup="status")
g1

g <- singleCellTK::plotPCA(indata_tmp2, colorBy = "nutrition", shape="HELMINTH", useAssay = "counts", runPCA = TRUE)
g <- g + geom_point(size = 3)
g


g1 + g
```

### Save
```{r}
ggsave(file.path(path, "pca_v6.svg"), plot=g, width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "pca_v6.png"), plot=g, width = 3.5, height = 3.5, pointsize = 5)

```

## UMAP {.tabset}

```{r }
assay_type = "log_counts_cpm"

set.seed(1)
umap_out <- umap(t(assay(indata,assay_type)))

embedding <- as.data.frame(umap_out$layout)
embedding$Class <- as.factor(indata$nutrition)
embedding$Helminth <- as.factor(indata$HELMINTH)
embedding$status <- as.factor(indata$status)

g <- ggplot(embedding, aes(x=V1, y=V2, color=Class, shape=Helminth, label = colnames(assay(indata,assay_type)))) + geom_point(size=3) + xlab("UMAP 1") + 
  ylab("UMAP 2") + 
  theme(plot.title = element_text(hjust = 0.5)) 

plot(g)
```
### Save
```{r}
ggsave(file.path(path, "umap_v6.svg"),plot=g, width = 3.5, height = 3.5, pointsize = 5)
ggsave(file.path(path, "umap_v6.png"),plot=g, width = 3.5, height = 3.5, pointsize = 5)

```


# Differential Expression Visit 6 {.tabset}
```{r}
indata <- se[,colData(se)$visit=="Visit_6"]
```

## Helminth vs no Helminth, control for nutrition {.tabset}
```{r }
visit <- "v6_"
path <- file.path("outs/differential_expression/visit_6/helPres_v_helAbs_con_nutrition")
#dir.create(path, recursive = T)
# matrix for differential expression
designMat <- model.matrix(~factor(HELMINTH) + factor(nutrition), data=colData(indata))

colnames(designMat)
colnames(designMat) <- c("Intercept", "Present", "well")

head(designMat)

fit <- lmFit(assay(indata, "log_counts_cpm"), designMat)

# Difference in expression between Failures and controls 

contrast.matrixNut<- makeContrasts(Present, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata)$subjtype)

new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
##shows spike near 0

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]

write.csv(sig_limmaResNut, file.path(path,paste0(visit, "helminth_con_nutrition_sig.csv")))
# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]

```



### 2 All genes datatable and heatmap {.tabset}
```{r }
DT::datatable(sig_limmaResNut, 
          options=list(scrollX=T,pAGELength=20),
          rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid") %>% group_by(Helminth) %>% arrange(Helminth, BMI)
mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(Helminth, Nutrition))

### annotation data for heatmap
#df=data.frame(Helminth=colData(indata)$"HELMINTH",
#              Nutrition=colData(indata)$"nutrition") 


ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))
g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )
g1 <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = F )
g + g1

```

number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

### save
```{r}
write.csv(sig_limmaResNut, file.path(path,paste0(visit, "helminth_con_nutrition_sig.csv")))
write.csv(sig_limmaResNut, file.path(ipa,paste0(visit, "helminth_con_nutrition_sig.csv")))

# save plot
svg(file.path(path, "helminth_con_nutrition_sig.svg"), width=7, height=7)
g
dev.off()
png(file.path(path, "helminth_con_nutrition_sig.png"), width=700, height=700)
g
dev.off()

# save plot
svg(file.path(path, "helminth_con_nutrition_sig_ordered.svg"), width=7, height=7)
g1
dev.off()
png(file.path(path, "helminth_con_nutrition_sig_ordered.png"), width=700, height=700)
g1
dev.off()

```


## Helminth vs no Helminth, mal-nourished only {.tabset}
```{r }
path <- file.path("outs/differential_expression/visit_6/mal_only_helPos_v_helAbs")
dir.create(path)
indata.mal <- indata[,colData(indata)$nutrition=="mal"]
# matrix for differential expression
designMat <- model.matrix(~factor(HELMINTH) , data=colData(indata.mal)) # with mal < 16; no Helminth negative

colnames(designMat)
colnames(designMat) <- c("Intercept", "Present")

head(designMat)

fit <- lmFit(assay(indata.mal, "log_counts_cpm"), designMat)

# Difference in expression between Failures and controls 

contrast.matrixNut<- makeContrasts(Present, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata)$subjtype)

new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)

new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)

##shows spike near 0

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]


# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]


```



### 2 All genes datatable and heatmap {.tabset}
```{r }
DT::datatable(sig_limmaResNut, 
          options=list(scrollX=T,pAGELength=20),
          rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH") 


ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))
g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )

write.csv(sig_limmaResNut, file.path(path, paste0(visit,"helminth_mal_only_sig.csv")))

# save plot
svg(file.path(path, "helminth_mal_only_sig.svg"), width=7, height=7)
g
dev.off()
png(file.path(path, "helminth_mal_only_sig.png"), width=700, height=700)
g
dev.off()

g
```

number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

## Helminth vs no Helminth, well-nourished only {.tabset}
```{r }
path <- file.path("outs/differential_expression", "visit_6", "well_only_helPos_v_helAbs")
dir.create(path)
indata.well <- indata[,colData(indata)$nutrition=="well"]
# matrix for differential expression
designMat <- model.matrix(~factor(HELMINTH) , data=colData(indata.well))

colnames(designMat)
colnames(designMat) <- c("Intercept", "Present")

head(designMat)

fit <- lmFit(assay(indata.well, "log_counts_cpm"), designMat)

# Difference in expression between Failures and controls 

contrast.matrixNut<- makeContrasts(Present, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata)$subjtype)

new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
#new_hist_padjval_100breaks
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
#new_hist_pval_100breaks
##shows spike near 0

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]


# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]


```



### 2 All genes datatable and heatmap {.tabset}
```{r }
DT::datatable(sig_limmaResNut, 
          options=list(scrollX=T,pAGELength=20),
          rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata.well[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation data for heatmap
df=data.frame(Helminth=colData(indata.well)$"HELMINTH",
              Nutrition=colData(indata.well)$"nutrition") 


ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))
g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )
g
write.csv(sig_limmaResNut, file.path(path, paste0(visit,"helminth_abs_vs_pres_well-only_sig.csv")))

# save plot
svg(file.path(path, "helminth_abs_vs_pres_well-only_sig.svg"), width=7, height=7)
g
dev.off()
png(file.path(path, "helminth_abs_vs_pres_well-only_sig.png"), width=700, height=700)
g
dev.off()

g
```

number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

## mal vs well, control for helminth {.tabset}
```{r }
path <- file.path("outs/differential_expression/visit_6","mal_vs_well_con_helminth")
dir.create(path)
# relevel so well is the intercept
colData(indata)$nutrition <- factor(colData(indata)$nutrition, levels=c("well","mal"))
levels(colData(indata)$nutrition)
# matrix for differential expression
designMat <- model.matrix(~nutrition + factor(HELMINTH) , data=colData(indata))

colnames(designMat)
colnames(designMat) <- c("Intercept", "mal", "Present")

head(designMat)

fit <- lmFit(assay(indata, "log_counts_cpm"), designMat)

# Difference in expression between well nourished and control for helminth status 

contrast.matrixNut<- makeContrasts(mal, levels = designMat)
fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
table(colData(indata)$nutrition)

# histograms of p values
new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
#new_hist_padjval_100breaks
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
#new_hist_pval_100breaks

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]


# top 1k genes by lowest p-value
top_1k_genes <- limmaResNut[1:1000,]

# top 25 high and low genes by logFC 
top_25_logfc <- sig_limmaResNut[order(sig_limmaResNut$logFC, decreasing=T),]
top_25_logfc <- top_25_logfc[1:25,]
top_25_neg_logfc <- sig_limmaResNut[order(sig_limmaResNut$logFC, decreasing = F),]
top_25_neg_logfc <- top_25_neg_logfc[1:25,]
```

### Volcano Plot
```{r}
#library(ggrepel)
mat = as.matrix(assay(indata,"log_counts_cpm"))

# dataframe to read for volcano plot
df <- limmaResNut %>% rownames_to_column(var="gene_symbol")

# label genes
label_genes <- c(rownames(top_25_logfc), rownames(top_25_neg_logfc))
# Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2fc respectively positive or negative)
df$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP"
df$diffexpressed[df$logFC > 0.6 & df$adj.P.Val < 0.05] <- "UP"
# if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"<br /><br /><br />
df$diffexpressed[df$logFC < -0.6 & df$adj.P.Val < 0.05] <- "DOWN"
unique(df$diffexpressed)
df <- df %>% mutate(delabel=ifelse(gene_symbol %in% label_genes, df$gene_symbol, NA))



unique()# Create a basic volcano plot
ggplot(data = df, aes(x = logFC, y = -log10(adj.P.Val), col = diffexpressed,  label = delabel)) +
  geom_vline(xintercept = c(-0.6, 0.6), col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') +
               scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), # to set the colours of our variable<br /><br /><br />
                     labels = c("Downregulated", "Not significant", "Upregulated")) +
  geom_point(size=2) +
               labs(color = 'Expression in Malnourished', #legend_title, 
       x = expression("log"[2]*"FC"), y = expression("-log"[10]*"(adjusted p-value)")) +
               ggtitle('Gene Expression at Visit 6') +
               geom_text_repel()
               

```

#### Save
```{r}
ggsave(file.path(path, "volcano.png"), width=7, height=7, units="in", dpi=300)
ggsave(file.path(path, "volcano.svg"), width=7, height=7, units="in", dpi=300)

```


### 2 All genes datatable and heatmap {.tabset}
```{r }
# datatable(sig_limmaResNut, 
#           options=list(scrollX=T,pAGELength=20),
#           rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))
mat = t(scale(t(mat)))

### annotation for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid",
              status=colData(indata)$"status") %>% arrange(BMI, Nutrition, Helminth)
mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(BMI,Nutrition,Helminth))


ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))

g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T )
g1 <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = F )

g + g1
```

number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)`

### save
```{r}
visit <- "v6_"
write.csv(sig_limmaResNut, file.path(path,paste0( visit, "mal_vs_well_con_helminth_sig.csv")))
write.csv(sig_limmaResNut, file.path(ipa,paste0( visit, "mal_vs_well_con_helminth_sig.csv")))
write.csv(top_25_logfc, file.path(path,paste0( visit, "top_25_logfc_mal_vs_well_con_helminth_sig.csv")))
write.csv(top_25_neg_logfc, file.path(path,paste0( visit, "top_25_neg_logfc_mal_vs_well_con_helminth_sig.csv")))


# save plot
svg(file.path(path, "mal_vs_well_con_helminth_sig.svg"), width=7, height=7)
g
dev.off()

png(file.path(path, "mal_vs_well_con_helminth_sig.png"), width=700, height=700)
g
dev.off()

# save plot
svg(file.path(path, "mal_vs_well_con_helminth_sig_ordered.svg"), width=7, height=7)
g1
dev.off()

png(file.path(path, "mal_vs_well_con_helminth_sig_ordered.png"), width=700, height=700)
g1
dev.off()
```

<!-- ## mal vs well, helminth positive{.tabset} -->
<!-- ```{r } -->
<!-- path <- file.path("outs/differential_expression", "visit_6", "mal_v_well_helPos_only") -->
<!-- dir.create(path) -->
<!-- # select only helminth positive -->
<!-- indata.pos <- indata[,colData(indata)$HELMINTH=="Present"] -->
<!-- # matrix for differential expression -->
<!-- designMat <- model.matrix(~factor(nutrition), data=colData(indata.pos)) -->

<!-- colnames(designMat) -->
<!-- colnames(designMat) <- c("Intercept", "well") -->

<!-- head(designMat) -->

<!-- fit <- lmFit(assay(indata.pos, "log_counts_cpm"), designMat) -->

<!-- # Difference in expression between well nourished and control for helminth status  -->

<!-- contrast.matrixNut<- makeContrasts(well, levels = designMat) -->
<!-- fitNut <- contrasts.fit(fit,contrast.matrixNut) -->
<!-- fitNut <- eBayes(fitNut) -->
<!-- limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P") -->


<!-- dim(limmaResNut[limmaResNut$adj.P.Val <0.05,]) -->
<!-- table(colData(indata.pos)$nutrition) -->

<!-- # histograms of p values -->
<!-- new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100) -->
<!-- #new_hist_padjval_100breaks -->
<!-- new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100) -->
<!-- #new_hist_pval_100breaks -->

<!-- sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,] -->
<!-- sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),] -->


<!-- # top 1k genes by lowest p-value -->
<!-- top_1k_genes <- limmaResNut[1:1000,] -->

<!-- ``` -->



<!-- ### 2 All genes datatable and heatmap {.tabset} -->
<!-- ```{r } -->
<!-- datatable(sig_limmaResNut,  -->
<!--           options=list(scrollX=T,pAGELength=20), -->
<!--           rownames = T) -->

<!-- ## Make a Heatmap of all genes -->
<!-- ### gene matrix, scaled across genes (genes as columns) -->
<!-- mat = as.matrix(assay(indata.pos[row.names(sig_limmaResNut),],"log_counts_cpm")) -->
<!-- mat = t(scale(t(mat))) -->

<!-- ### annotation data for heatmap -->
<!-- df=data.frame(Nutrition=colData(indata.pos)$"nutrition", Helminths=colData(indata.pos)$"HELMINTH")  -->


<!-- ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow"))) -->

<!-- g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T ) -->
<!-- write.csv(sig_limmaResNut, file.path(path,paste0(visit,"mal_v_well_helminthPos_sig.csv"))) -->
<!-- # save plot -->
<!-- svg(file.path(path, "mal_v_well_helminthPos_sig.svg"), width=7, height=7) -->
<!-- g -->
<!-- dev.off() -->
<!-- png(file.path(path, "mal_v_well_helminthPos_sig.png"), width=700, height=700) -->
<!-- g -->
<!-- dev.off() -->
<!-- g -->
<!-- ``` -->

<!-- number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)` -->

<!-- ## mal vs well, helminth negative {.tabset} -->
<!-- ```{r } -->
<!-- visit <- "visit_6" -->
<!-- diffex_test <- "mal_v_well_helAbs_only" -->
<!-- path <- file.path("outs/differential_expression","visit_6", "mal_v_well_helAbs_only") -->
<!-- dir.create(path) -->
<!-- indata.neg <- indata[,colData(indata)$HELMINTH=="Absent"] -->
<!-- # matrix for differential expression -->
<!-- designMat <- model.matrix(~factor(nutrition), data=colData(indata.neg)) -->

<!-- colnames(designMat) -->
<!-- colnames(designMat) <- c("Intercept", "well") -->

<!-- head(designMat) -->

<!-- fit <- lmFit(assay(indata.neg, "log_counts_cpm"), designMat) -->

<!-- # Difference in expression between well nourished and control for helminth status  -->

<!-- contrast.matrixNut<- makeContrasts(well, levels = designMat) -->
<!-- fitNut <- contrasts.fit(fit,contrast.matrixNut) -->
<!-- fitNut <- eBayes(fitNut) -->
<!-- limmaResNut <- topTable(fitNut, adjust.method = "BH", n = Inf, sort.by = "P") -->


<!-- dim(limmaResNut[limmaResNut$adj.P.Val <0.05,]) -->
<!-- table(colData(indata.neg)$nutrition) -->

<!-- # histograms of p values -->
<!-- new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100) -->
<!-- #new_hist_padjval_100breaks -->
<!-- new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100) -->
<!-- #new_hist_pval_100breaks -->

<!-- sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.05,] -->
<!-- sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),] -->

<!-- # top 1k genes by lowest p-value -->
<!-- top_1k_genes <- limmaResNut[1:1000,] -->

<!-- ``` -->



<!-- ### 2 All genes datatable and heatmap {.tabset} -->
<!-- ```{r } -->
<!-- datatable(sig_limmaResNut,  -->
<!--           options=list(scrollX=T,pAGELength=20), -->
<!--           rownames = T) -->

<!-- ## Make a Heatmap of all genes -->
<!-- ### gene matrix, scaled across genes (genes as columns) -->
<!-- mat = as.matrix(assay(indata.neg[row.names(sig_limmaResNut),],"log_counts_cpm")) -->
<!-- mat = t(scale(t(mat))) -->

<!-- ### annotation data for heatmap -->
<!-- df=data.frame(Nutrition=colData(indata.neg)$"nutrition", Helminths=colData(indata.neg)$"HELMINTH")  -->


<!-- ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow"))) -->

<!-- g <- Heatmap(mat,show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T ) -->

<!-- # save plot -->
<!-- write.csv(sig_limmaResNut, file.path(path, paste0(visit,"mal_v_well_helminthNeg_sig.csv"))) -->
<!-- # save plot -->
<!-- svg(file.path(path, "mal_v_well_helminthNeg_sig.svg"), width=7, height=7) -->
<!-- g -->
<!-- dev.off() -->
<!-- png(file.path(path, "mal_v_well_helminthNeg_sig.png"), width=700, height=700) -->
<!-- g -->
<!-- dev.off() -->
<!-- g -->
<!-- ``` -->

<!-- number of genes of p.adj<0.05=`r nrow(sig_limmaResNut)` -->


# Differential Expression V1 vs V6
```{r}
# path for V1 vs V6 effects due to nutrition
status <- "all"
path <- file.path("outs/differential_expression/v1_v_v6/", status)
# path for V1 vs V6 effects due to helminth infection
status <- "all_Helminth_effects"
path <- file.path("outs/differential_expression/v1_v_v6/", status)
```


## All mal V1 vs V6
```{r}
# subset
indata <- se[,colData(se)$nutrition=="mal"]
df <- as.data.frame(colData(indata)) %>% group_by(ID1) %>% filter( n() > 1 )
indata <- indata[,colData(indata)$ID1 %in% df$ID1]
#indata$HELMINTH <- as.factor(indata$HELMINTH)
# status
status="mal"
```

### significant genes
```{r}
indata <- se
status <- "all"
path <- file.path("outs/differential_expression/v1_v_v6/", status)
#dir.create(path, recursive = T)
# matrix for differential expression
#designMat <- model.matrix(~factor(visit) + factor(HELMINTH), data=colData(indata))
table(colData(indata)$nutrition, colData(indata)$visit, colData(indata)$HELMINTH)

colData(indata)$nutrition <- factor(colData(indata)$nutrition, levels=c("well", "mal"))

#designMat <- model.matrix(~factor(ID1) + factor(HELMINTH)  + nutrition*factor(visit) , data=colData(indata))
designMat <- model.matrix(~ factor(HELMINTH)  + nutrition*factor(visit) , data=colData(indata))

designMat <- model.matrix(~ factor(HELMINTH) + nutrition*factor(visit) + ID1 , data=colData(indata))

#design <- model.matrix(~ Sample + Helminth + Group * Visit, data=coldata)

colnames(designMat)
colnames(designMat) <- gsub(pattern="factor\\(ID1\\)", replacement =  "", colnames(designMat))
colnames(designMat) <- gsub(pattern="factor\\(visit\\)", replacement =  "", colnames(designMat))
colnames(designMat) <- gsub(pattern="factor\\(HELMINTH\\)", replacement =  "", colnames(designMat))
colnames(designMat) <- gsub(pattern="nutritionmal", replacement =  "mal", colnames(designMat))
colnames(designMat) <- gsub(pattern="\\(Intercept\\)", replacement =  "Visit_1_wellnourished", colnames(designMat))
colnames(designMat) <- gsub(pattern="mal:Visit_6", replacement =  "Visit_6_mal", colnames(designMat))
head(designMat)



fit <- lmFit(assay(indata, "log_counts_cpm"), designMat)

# Difference in expression between Failures and controls 
### pseudocode
#1. what's the effect of nutritional supplement? 
### contrast on visit_6
#2. Do malnourished visit 6 look like well-nourished people?
### is visit_6_mal=average(well_v1, well_v6)=
###### intercept + visit_6    = 0.5*(intercept + well) + 0.5*(intercept + well + visit_6 + well:visit_6)

######## visit_6=0.5(well) + 0.5(well + visit_6 + well:visit_6)
######## visit_6=well + 0.5(visit_6 + well:visit_6)
######## 0.5(visit_6)=well + 0.5(well:visit_6)
######## 0.5(visit_6)-well - 0.5(well:visit_6)=0
# contrast needs to be:(visit_6)-2*well - (well:visit_6)=0

#### we need intercept + visit_6 effect
#### for well_v1 we need intercept + well
#### well_v6 is intercept + well + visit_6 + well:visit_6



contrast.matrixNut<- makeContrasts(Visit_6, levels = designMat) # visit 1 vs visit in well?
contrast.matrixNut<- makeContrasts(Visit_6 - 2*mal - Visit_6_mal, levels = designMat) # is visit 6 Well different than avg of visit 1/6 Mal?

#visit_6 mal = Visit_1_wellnourished + mal + Visit_6_mal
#avg(visit1 well + visit6 well) = 1/2 * (Visit_1_wellnourished + Visit_1_wellnourished + Visit6) = Visit_1_wellnourished + 0.5 Visit 6
## visit_6 mal - avg(visit1 well + visit6 well) = 0 
##  mal + Visit_6_mal - 0.5 Visit 6 = 0
contrast.matrixNut<- makeContrasts(2*mal + 2*Visit_6_mal - Visit_6, levels = designMat) # is visit 6 Mal different than avg of visit 1/6 Well?
# # # ##############
contrast.matrixNut<- makeContrasts(Visit_6 + Visit_6_mal, levels = designMat) # is visit 6 mal different than Visit 1 mal?

fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, n = Inf, adjust.method = "BH", sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
dim(limmaResNut[limmaResNut$adj.P.Val <0.10,])
dim(limmaResNut[limmaResNut$P.Value <0.005,])
hist(limmaResNut$P.Value)
table(colData(indata)$subjtype)

new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
##shows spike near 0

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.1,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]
sig_limmaResNut2k <- sig_limmaResNut[1:2000,]
```
### 2 All genes datatable and heatmap {.tabset}
```{r }
# DT::datatable(sig_limmaResNut, 
#           options=list(scrollX=T,pAGELength=20),
#           rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(limmaResNut[limmaResNut$adj.P.Val <0.1,]),],"log_counts_cpm"))
mat = t(scale(t(mat)))


### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              status=colData(indata)$"status",
              Visit=colData(indata)$"visit",
              ID1=colData(indata)$"ID1",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid") %>% dplyr::arrange(Visit,status )


mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(status, Visit, BMI))


# remove ID from annotation
#df$ID <- NULL

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminth=c("Present"="Black", "Absent"="Yellow"), Visit=c("Visit_1"="Grey", "Visit_6"="Turquoise")))
g <- Heatmap(mat, show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T, use_raster = T )
g1 <- Heatmap(mat, show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = F, use_raster = T )
g + g1


```
#### save
```{r}

path <- file.path("outs/differential_expression/v1_v_v6/all")

png(file.path(path, "v1_vs_v6_nutrition_by_visit_sig_genes_ordered.png"), width=7, height=7,res=300, units = "in")
g1
dev.off()
svg(file.path(path, "v1_vs_v6_nutrition_by_visit_sig_genes_ordered.svg"), width=7, height=7)
g1
dev.off()

write.csv(sig_limmaResNut, file.path(path, "visit_by_nutrition_interaction_plus_helminth.csv"))


```

### dimredu
```{r}
# Get PCA data
mat <- assay(indata, "log_counts_cpm")  # Variance Stabilization
#mat <- t(scale(t(mat)))  # Variance Stabilization

# Perform PCA on voom-transformed expression
pca <- prcomp(t(mat), scale.=TRUE)  

# Extract PCA variance explained
percentVar <- round(100 * summary(pca)$importance[2, 1:2], 1)

# Create a dataframe for plotting
coldata <- as.data.frame(colData(indata))
pcaData <- data.frame(
  Sample = coldata$ID1,
  PC1 = pca$x[,1],
  PC2 = pca$x[,2],
  Nutrition = coldata$nutrition,
  Visit = coldata$visit,
  Helminth = coldata$HELMINTH
)
pcaData$Visit <- factor(pcaData$Visit, levels = c("Visit_1", "Visit_6"))
# Create a paired dataset for arrows
pca_pairs <- merge(
  pcaData[which(pcaData$Visit == "Visit_1"), ], 
  pcaData[which(pcaData$Visit == "Visit_6"), ], 
  by="Sample", suffixes=c("_before", "_after"),
  incomparables = NA
)


pca_before <- pcaData[pcaData$Visit == "Visit_1", ]
pca_after <- pcaData[pcaData$Visit == "Visit_6", ]

pca_pairs <- data.frame(
  Sample = pca_before$Sample,
  PC1_before = pca_before$PC1,
  PC2_before = pca_before$PC2,
  PC1_after = pca_after$PC1[match(pca_before$Sample, pca_after$Sample)],
  PC2_after = pca_after$PC2[match(pca_before$Sample, pca_after$Sample)],
  Helminth = pca_before$Helminth,
  Nutrition=pca_before$Nutrition
)

# Compute centroid (mean PC1 and PC2) for Well-Nourished at each visit
well_centroids <- aggregate(cbind(PC1, PC2) ~ Visit, 
                            data=pcaData[pcaData$Nutrition == "well", ], mean)

# PCA plot with paired arrows
ggplot(pcaData, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=Nutrition, shape=Visit), size=3) +
  geom_segment(inherit.aes = F,
               data=pca_pairs,
               aes(x=PC1_before, y=PC2_before, xend=PC1_after, yend=PC2_after, group=Sample, color=Helminth),
               arrow=arrow(length=unit(0.005,"cm")), alpha=0.4, linewidth=0.5) +  # Arrows for paired samples
  scale_color_manual(values=c("Absent"="grey",
                              "Present"="orange",
                              "mal"="red",
                              "well"="blue")) +  # Helminth color coding
  labs(title="PCA Plot: Normalized Data",
       x=paste0("PC1 (", percentVar[1], "% variance)"),
       y=paste0("PC2 (", percentVar[2], "% variance)"),
       color="Status") +
  theme_minimal()

### try PCoA's????

# PCA plot with paired arrows
ggplot(pcaData, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=Nutrition, shape=Visit), size=3) +
  annotate("point",x=well_centroids$PC1[2], y=well_centroids$PC2[2], colour="black", size=5)  +           
  geom_segment(inherit.aes = F, 
               data=pca_pairs, 
               aes(x=PC1_before, y=PC2_before, xend=PC1_after, yend=PC2_after, group=Sample, color=Nutrition), 
               arrow=arrow(length = unit(0.05, "cm"), type = "closed"), alpha=0.9, linewidth=0.75) +  # Arrows for paired samples
  scale_color_manual(values=c("mal"="red", 
                              "well"="blue")) +  # Helminth color coding
  labs(title="PCA Plot: Normalized Data",
       x=paste0("PC1 (", percentVar[1], "% variance)"),
       y=paste0("PC2 (", percentVar[2], "% variance)"),
       color="Status") +
  theme_minimal()
```
### save
```{r}
path <- file.path("outs/differential_expression/v1_v_v6/all")

ggsave(file.path(path, "euclidean_distance_PCA.png"), width=7, height=5)
ggsave(file.path(path, "euclidean_distance_PCA.svg"), width=7, height=5)


```


```{r}


pcaData <- pcaData %>%
  left_join(well_centroids, by="Visit", suffix=c("", "_well")) %>%
  mutate(Distance_to_Well = sqrt((PC1 - PC1_well)^2 + (PC2 - PC2_well)^2))
```

```{r}
ggplot(pcaData[pcaData$Nutrition == "mal", ], aes(x=Visit, y=Distance_to_Well)) +
  geom_boxplot(aes(fill=Visit)) +
  geom_point(position=position_jitter(width=0.2), alpha=0.5) +
  labs(title="Distance from Malnourished to Well-Nourished Centroid",
       y="Euclidean Distance", x="Visit") +
  theme_minimal()

ggplot(pcaData, aes(x=Visit, y=Distance_to_Well)) +
  geom_boxplot(aes(fill=Nutrition),position="dodge")  +
  labs(title="Distance from Malnourished to Well-Nourished Centroid",
       y="Euclidean Distance", x="Visit") +
  theme_minimal()

```
### save 
```{r}
path <- file.path("outs/differential_expression/v1_v_v6/all")

ggsave(file.path(path, "euclidean_distance_PCA_boxplot.png"), width=5, height=5)
ggsave(file.path(path, "euclidean_distance_PCA_boxplot.svg"), width=5, height=5)

```



```{r}
malnourished_distances <- pcaData %>%
  filter(Nutrition == "mal") %>%
  select(Sample, Visit, Distance_to_Well) %>%
  spread(Visit, Distance_to_Well) %>%
  filter(!is.na(Visit_1) & !is.na(Visit_6))

# Paired t-test
t.test(malnourished_distances$Visit_1, malnourished_distances$Visit_6, paired=TRUE)
```
```{r}
malnourished_distances <- pcaData %>% 
  select(Sample,Nutrition, Visit, Distance_to_Well) %>%
  spread(Visit, Distance_to_Well) %>%
  filter(!is.na(Visit_1) & !is.na(Visit_6))

df <- filter(pcaData, Sample %in% malnourished_distances$Sample)

aov_result <- aov(Distance_to_Well ~ Nutrition * Visit + Error(Sample/Visit), data = df)
summary(aov_result)

```
```{r}
df$grouping <- interaction(df$Nutrition, df$Visit)

pairwise.t.test(df$Distance_to_Well, df$grouping, paired = FALSE, p.adjust.method = "bonferroni")

```


### save
```{r}
## save genes
write.csv(sig_limmaResNut, file.path(path, paste0(diffex_test,"_",status,"_sig.csv")))
write.csv(sig_limmaResNut2k, file.path(ipa, paste0(diffex_test,"_",status,"_sig.csv")))


# save plot
svg(file.path(path, paste0(status,"_sig.svg")), width=7, height=7)
g
dev.off()
png(file.path(path, paste0(status,"_sig.png")), width=700, height=700)
g
dev.off()

# save plot
svg(file.path(path, paste0(status,"_ordered_sig.svg")), width=7, height=7)
g1
dev.off()
png(file.path(path, paste0(status,"_ordered_sig.png")), width=700, height=700)
g1
dev.off()

mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))


```


### 2 All genes datatable and heatmap log-log {.tabset}
```{r }
# DT::datatable(sig_limmaResNut, 
#           options=list(scrollX=T,pAGELength=20),
#           rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))

mat.1 <- as.data.frame(mat) 

# list of ID
ids <- unique(colData(indata)$"ID1")
m <- sapply(ids, function(x) mat.1[,paste0(x, "_VISIT_6")] - mat.1[,paste0(x, "_VISIT_1")])
rownames(m) <- rownames(mat.1)

### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              Visit=colData(indata)$"visit",
              ID1=colData(indata)$"ID1",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid") %>% dplyr::arrange(BMI, Visit, Nutrition, ID1)

mat = t(scale(t(mat)))
mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(Helminth, BMI, Nutrition))


# remove ID from annotation
#df$ID <- NULL

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminth=c("Present"="Black", "Absent"="Yellow")))
g <- Heatmap(mat, show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T, use_raster = T )
g1 <- Heatmap(mat, show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = F, use_raster = T )
g + g1


```


```{r}
## save genes
# write.csv(sig_limmaResNut, file.path(path, paste0(diffex_test,"_",status,"_sig.csv")))
# write.csv(sig_limmaResNut, file.path(ipa, paste0(diffex_test,"_",status,"_sig.csv")))


# save plot
svg(file.path(path, paste0(status,"_log-log_sig.svg")), width=7, height=7)
g
dev.off()
png(file.path(path, paste0(status,"_log-log_sig.png")), width=700, height=700)
g
dev.off()

# save plot
svg(file.path(path, paste0(status,"_ordered_log-log_sig.svg")), width=7, height=7)
g1
dev.off()
png(file.path(path, paste0(status,"_ordered_log-log_sig.png")), width=700, height=700)
g1
dev.off()

```


## All V1 vs V6 - Helminth effects
```{r}
# subset
#ndata <- se
# df <- as.data.frame(colData(indata)) %>% group_by(ID1) %>% filter( n() > 1 )
# indata <- indata[,colData(indata)$ID1 %in% df$ID1]
# indata$HELMINTH <- as.factor(indata$HELMINTH)
# status
#status="all_helminth_effects"
```

### significant 
```{r}
indata <- se
status <- "all_Helminth_effects"
path <- file.path("outs/differential_expression/v1_v_v6/", status)
#dir.create(path, recursive = T)
# matrix for differential expression
#designMat <- model.matrix(~factor(visit) + factor(HELMINTH), data=colData(indata))
table(colData(indata)$nutrition, colData(indata)$visit, colData(indata)$HELMINTH)

colData(indata)$nutrition <- factor(colData(indata)$nutrition, levels=c("well", "mal"))

#designMat <- model.matrix(~factor(ID1) + factor(HELMINTH)  + nutrition*factor(visit) , data=colData(indata))
designMat <- model.matrix(~ factor(HELMINTH)  + nutrition*factor(visit) , data=colData(indata))

designMat <- model.matrix(~ factor(HELMINTH) + nutrition*factor(visit) + ID1 , data=colData(indata))

#design <- model.matrix(~ Sample + Helminth + Group * Visit, data=coldata)

colnames(designMat)
colnames(designMat) <- gsub(pattern="factor\\(ID1\\)", replacement =  "", colnames(designMat))
colnames(designMat) <- gsub(pattern="factor\\(visit\\)", replacement =  "", colnames(designMat))
colnames(designMat) <- gsub(pattern="factor\\(HELMINTH\\)", replacement =  "", colnames(designMat))
colnames(designMat) <- gsub(pattern="nutritionmal", replacement =  "mal", colnames(designMat))
colnames(designMat) <- gsub(pattern="\\(Intercept\\)", replacement =  "Visit_1_wellnourished", colnames(designMat))
colnames(designMat) <- gsub(pattern="mal:Visit_6", replacement =  "Visit_6_mal", colnames(designMat))
head(designMat)



fit <- lmFit(assay(indata, "log_counts_cpm"), designMat)

# Difference in expression between Failures and controls 
### pseudocode
#1. what's the effect of nutritional supplement? 
### contrast on visit_6
#2. Do malnourished visit 6 look like well-nourished people?
### is visit_6_mal=average(well_v1, well_v6)=
###### intercept + visit_6    = 0.5*(intercept + well) + 0.5*(intercept + well + visit_6 + well:visit_6)

######## visit_6=0.5(well) + 0.5(well + visit_6 + well:visit_6)
######## visit_6=well + 0.5(visit_6 + well:visit_6)
######## 0.5(visit_6)=well + 0.5(well:visit_6)
######## 0.5(visit_6)-well - 0.5(well:visit_6)=0
# contrast needs to be:(visit_6)-2*well - (well:visit_6)=0

#### we need intercept + visit_6 effect
#### for well_v1 we need intercept + well
#### well_v6 is intercept + well + visit_6 + well:visit_6



contrast.matrixNut<- makeContrasts(Visit_6, levels = designMat) # visit 1 vs visit in well?
contrast.matrixNut<- makeContrasts(Visit_6 - 2*mal - Visit_6_mal, levels = designMat) # is visit 6 Well different than avg of visit 1/6 Mal?

#visit_6 mal = Visit_1_wellnourished + mal + Visit_6_mal
#avg(visit1 well + visit6 well) = 1/2 * (Visit_1_wellnourished + Visit_1_wellnourished + Visit6) = Visit_1_wellnourished + 0.5 Visit 6
## visit_6 mal - avg(visit1 well + visit6 well) = 0 
##  mal + Visit_6_mal - 0.5 Visit 6 = 0
contrast.matrixNut<- makeContrasts(2*mal + 2*Visit_6_mal - Visit_6, levels = designMat) # is visit 6 Mal different than avg of visit 1/6 Well?
# # # ##############
contrast.matrixNut<- makeContrasts(Visit_6 + Visit_6_mal, levels = designMat) # is visit 6 mal different than Visit 1 mal?

fitNut <- contrasts.fit(fit,contrast.matrixNut)
fitNut <- eBayes(fitNut)
limmaResNut <- topTable(fitNut, n = Inf, adjust.method = "BH", sort.by = "P")


dim(limmaResNut[limmaResNut$adj.P.Val <0.05,])
dim(limmaResNut[limmaResNut$adj.P.Val <0.10,])
dim(limmaResNut[limmaResNut$P.Value <0.005,])
hist(limmaResNut$P.Value)
table(colData(indata)$subjtype)

new_hist_padjval_100breaks <- hist(limmaResNut$adj.P.Val, breaks=100)
new_hist_pval_100breaks <- hist(limmaResNut$P.Value, breaks=100)
##shows spike near 0

sig_limmaResNut <- limmaResNut[limmaResNut$adj.P.Val <0.1,]
sig_limmaResNut <- sig_limmaResNut[order(sig_limmaResNut$adj.P.Val),]
sig_limmaResNut2k <- sig_limmaResNut[1:2000,]
```
### 2 All genes datatable and heatmap {.tabset}
```{r }
# DT::datatable(sig_limmaResNut, 
#           options=list(scrollX=T,pAGELength=20),
#           rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(limmaResNut[limmaResNut$adj.P.Val <0.1,]),],"log_counts_cpm"))
mat = t(scale(t(mat)))


### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              status=colData(indata)$"status",
              Visit=colData(indata)$"visit",
              ID1=colData(indata)$"ID1",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid") %>% dplyr::arrange(Visit,status )


mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(status, Visit, BMI))


# remove ID from annotation
#df$ID <- NULL

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminth=c("Present"="Black", "Absent"="Yellow"), Visit=c("Visit_1"="Grey", "Visit_6"="Turquoise")))
g <- Heatmap(mat, show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T, use_raster = T )
g1 <- Heatmap(mat, show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = F, use_raster = T )
g + g1


```
#### save
```{r}

path <- file.path("outs/differential_expression/v1_v_v6/all")

png(file.path(path, "v1_vs_v6_nutrition_by_visit_sig_genes_ordered.png"), width=7, height=7,res=300, units = "in")
g1
dev.off()
svg(file.path(path, "v1_vs_v6_nutrition_by_visit_sig_genes_ordered.svg"), width=7, height=7)
g1
dev.off()

write.csv(sig_limmaResNut, file.path(path, "visit_by_nutrition_interaction_plus_helminth.csv"))


```

### dimredu
```{r}
# Get PCA data
mat <- assay(indata, "log_counts_cpm")  # Variance Stabilization
#mat <- t(scale(t(mat)))  # Variance Stabilization

# Perform PCA on voom-transformed expression
pca <- prcomp(t(mat), scale.=TRUE)  

# Extract PCA variance explained
percentVar <- round(100 * summary(pca)$importance[2, 1:2], 1)

# Create a dataframe for plotting
coldata <- as.data.frame(colData(indata))
pcaData <- data.frame(
  Sample = coldata$ID1,
  PC1 = pca$x[,1],
  PC2 = pca$x[,2],
  Nutrition = coldata$nutrition,
  Visit = coldata$visit,
  Helminth = coldata$HELMINTH
)
pcaData$Visit <- factor(pcaData$Visit, levels = c("Visit_1", "Visit_6"))
# Create a paired dataset for arrows
pca_pairs <- merge(
  pcaData[which(pcaData$Visit == "Visit_1"), ], 
  pcaData[which(pcaData$Visit == "Visit_6"), ], 
  by="Sample", suffixes=c("_before", "_after"),
  incomparables = NA
)


pca_before <- pcaData[pcaData$Visit == "Visit_1", ]
pca_after <- pcaData[pcaData$Visit == "Visit_6", ]

pca_pairs <- data.frame(
  Sample = pca_before$Sample,
  PC1_before = pca_before$PC1,
  PC2_before = pca_before$PC2,
  PC1_after = pca_after$PC1[match(pca_before$Sample, pca_after$Sample)],
  PC2_after = pca_after$PC2[match(pca_before$Sample, pca_after$Sample)],
  Helminth = pca_before$Helminth,
  Nutrition=pca_before$Nutrition
)

# Compute centroid (mean PC1 and PC2) for Well-Nourished at each visit
well_centroids <- aggregate(cbind(PC1, PC2) ~ Visit, 
                            data=pcaData[pcaData$Nutrition == "well", ], mean)

# PCA plot with paired arrows
ggplot(pcaData, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=Nutrition, shape=Visit), size=3) +
  geom_segment(inherit.aes = F,
               data=pca_pairs,
               aes(x=PC1_before, y=PC2_before, xend=PC1_after, yend=PC2_after, group=Sample, color=Helminth),
               arrow=arrow(length=unit(0.005,"cm")), alpha=0.4, linewidth=0.5) +  # Arrows for paired samples
  scale_color_manual(values=c("Absent"="grey",
                              "Present"="orange",
                              "mal"="red",
                              "well"="blue")) +  # Helminth color coding
  labs(title="PCA Plot: Normalized Data",
       x=paste0("PC1 (", percentVar[1], "% variance)"),
       y=paste0("PC2 (", percentVar[2], "% variance)"),
       color="Status") +
  theme_minimal()

### try PCoA's????

# PCA plot with paired arrows
ggplot(pcaData, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=Nutrition, shape=Visit), size=3) +
  annotate("point",x=well_centroids$PC1[2], y=well_centroids$PC2[2], colour="black", size=5)  +           
  geom_segment(inherit.aes = F, 
               data=pca_pairs, 
               aes(x=PC1_before, y=PC2_before, xend=PC1_after, yend=PC2_after, group=Sample, color=Nutrition), 
               arrow=arrow(length = unit(0.05, "cm"), type = "closed"), alpha=0.9, linewidth=0.75) +  # Arrows for paired samples
  scale_color_manual(values=c("mal"="red", 
                              "well"="blue")) +  # Helminth color coding
  labs(title="PCA Plot: Normalized Data",
       x=paste0("PC1 (", percentVar[1], "% variance)"),
       y=paste0("PC2 (", percentVar[2], "% variance)"),
       color="Status") +
  theme_minimal()
```
### save
```{r}
path <- file.path("outs/differential_expression/v1_v_v6/all")

ggsave(file.path(path, "euclidean_distance_PCA.png"), width=7, height=5)
ggsave(file.path(path, "euclidean_distance_PCA.svg"), width=7, height=5)


```


```{r}


pcaData <- pcaData %>%
  left_join(well_centroids, by="Visit", suffix=c("", "_well")) %>%
  mutate(Distance_to_Well = sqrt((PC1 - PC1_well)^2 + (PC2 - PC2_well)^2))
```

```{r}
ggplot(pcaData[pcaData$Nutrition == "mal", ], aes(x=Visit, y=Distance_to_Well)) +
  geom_boxplot(aes(fill=Visit)) +
  geom_point(position=position_jitter(width=0.2), alpha=0.5) +
  labs(title="Distance from Malnourished to Well-Nourished Centroid",
       y="Euclidean Distance", x="Visit") +
  theme_minimal()

ggplot(pcaData, aes(x=Visit, y=Distance_to_Well)) +
  geom_boxplot(aes(fill=Nutrition),position="dodge")  +
  labs(title="Distance from Malnourished to Well-Nourished Centroid",
       y="Euclidean Distance", x="Visit") +
  theme_minimal()

```
### save 
```{r}
path <- file.path("outs/differential_expression/v1_v_v6/all")

ggsave(file.path(path, "euclidean_distance_PCA_boxplot.png"), width=5, height=5)
ggsave(file.path(path, "euclidean_distance_PCA_boxplot.svg"), width=5, height=5)

```



```{r}
malnourished_distances <- pcaData %>%
  filter(Nutrition == "mal") %>%
  select(Sample, Visit, Distance_to_Well) %>%
  spread(Visit, Distance_to_Well) %>%
  filter(!is.na(Visit_1) & !is.na(Visit_6))

# Paired t-test
t.test(malnourished_distances$Visit_1, malnourished_distances$Visit_6, paired=TRUE)
```
```{r}
malnourished_distances <- pcaData %>% 
  select(Sample,Nutrition, Visit, Distance_to_Well) %>%
  spread(Visit, Distance_to_Well) %>%
  filter(!is.na(Visit_1) & !is.na(Visit_6))

df <- filter(pcaData, Sample %in% malnourished_distances$Sample)

aov_result <- aov(Distance_to_Well ~ Nutrition * Visit + Error(Sample/Visit), data = df)
summary(aov_result)

```
```{r}
df$grouping <- interaction(df$Nutrition, df$Visit)

pairwise.t.test(df$Distance_to_Well, df$grouping, paired = FALSE, p.adjust.method = "bonferroni")

```


### save
```{r}
## save genes
write.csv(sig_limmaResNut, file.path(path, paste0(diffex_test,"_",status,"_sig.csv")))
write.csv(sig_limmaResNut2k, file.path(ipa, paste0(diffex_test,"_",status,"_sig.csv")))


# save plot
svg(file.path(path, paste0(status,"_sig.svg")), width=7, height=7)
g
dev.off()
png(file.path(path, paste0(status,"_sig.png")), width=700, height=700)
g
dev.off()

# save plot
svg(file.path(path, paste0(status,"_ordered_sig.svg")), width=7, height=7)
g1
dev.off()
png(file.path(path, paste0(status,"_ordered_sig.png")), width=700, height=700)
g1
dev.off()

mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))


```


### 2 All genes datatable and heatmap log-log {.tabset}
```{r }
# DT::datatable(sig_limmaResNut, 
#           options=list(scrollX=T,pAGELength=20),
#           rownames = T)

## Make a Heatmap of all genes
### gene matrix, scaled across genes (genes as columns)
mat = as.matrix(assay(indata[row.names(sig_limmaResNut),],"log_counts_cpm"))

mat.1 <- as.data.frame(mat) 

# list of ID
ids <- unique(colData(indata)$"ID1")
m <- sapply(ids, function(x) mat.1[,paste0(x, "_VISIT_6")] - mat.1[,paste0(x, "_VISIT_1")])
rownames(m) <- rownames(mat.1)

### annotation data for heatmap
df=data.frame(Helminth=colData(indata)$"HELMINTH",
              Nutrition=colData(indata)$"nutrition",
              Visit=colData(indata)$"visit",
              ID1=colData(indata)$"ID1",
              BMI=colData(indata)$"BMI",
              ID=colData(indata)$"subjid") %>% dplyr::arrange(BMI, Visit, Nutrition, ID1)

mat = t(scale(t(mat)))
mat <- as.matrix(relocate(as.data.frame(mat), df$ID))

df <- df %>% select(c(Helminth, BMI, Nutrition))


# remove ID from annotation
#df$ID <- NULL

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminth=c("Present"="Black", "Absent"="Yellow")))
g <- Heatmap(mat, show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = T, use_raster = T )
g1 <- Heatmap(mat, show_row_names=F,show_column_names = F, top_annotation = ha, cluster_columns = F, use_raster = T )
g + g1


```


```{r}
## save genes
# write.csv(sig_limmaResNut, file.path(path, paste0(diffex_test,"_",status,"_sig.csv")))
# write.csv(sig_limmaResNut, file.path(ipa, paste0(diffex_test,"_",status,"_sig.csv")))


# save plot
svg(file.path(path, paste0(status,"_log-log_sig.svg")), width=7, height=7)
g
dev.off()
png(file.path(path, paste0(status,"_log-log_sig.png")), width=700, height=700)
g
dev.off()

# save plot
svg(file.path(path, paste0(status,"_ordered_log-log_sig.svg")), width=7, height=7)
g1
dev.off()
png(file.path(path, paste0(status,"_ordered_log-log_sig.png")), width=700, height=700)
g1
dev.off()

```


# Pathway analysis
```{r}

```

## IPA, MsigDB, EnrichR
### send the list of differentially expressed genes to a database using any of these tools and get results


# Gene set enrichment analysis {.tabset}
## TBsignatureProfiler {.tabset}
```{r, echo=F}
signatureBoxplot=function (inputData, annotationData, signatureColNames, annotationColName, 
                           name = "Signatures", scale = FALSE, violinPlot = FALSE, 
                           includePoints = TRUE, notch = FALSE, rotateLabels = FALSE, 
                           nrow = NULL, ncol = NULL, fill_colors = NULL) 
{
  if (methods::is(inputData, "SummarizedExperiment")) {
    if (any(duplicated(signatureColNames))) {
      stop("Duplicate signature column name is not supported.")
    }
    if (!all(signatureColNames %in% colnames(SummarizedExperiment::colData(inputData)))) {
      stop("Signature column name not found in inputData.")
    }
    if (!all(annotationColName %in% colnames(SummarizedExperiment::colData(inputData)))) {
      stop("Annotation column name not found in inputData.")
    }
    annotationData <- data.frame(SummarizedExperiment::colData(inputData)[, 
                                                                          annotationColName, drop = FALSE])
    inputData <- data.frame(SummarizedExperiment::colData(inputData)[, 
                                                                     signatureColNames, drop = FALSE])
  }
  else {
    if (ncol(annotationData) != 1) {
      stop("annotationData must have only one column.")
    }
    annotationColName <- colnames(annotationData)
  }
  if (length(annotationColName) != 1) {
    stop("You must specify a single annotation column name to color boxplots by.")
  }
  if (!is.factor(annotationData[, 1])) {
    annotationData[, 1] <- as.factor(annotationData[, 1])
  }
  n <- length(levels(annotationData[, 1]))
  if (n > 9) {
    stop("Too many levels in the annotation data. The boxplot can contain a maximum of 9 levels")
  }
  if (nrow(annotationData) == nrow(inputData)) {
    if (!all(rownames(annotationData) == rownames(inputData))) {
      stop("Annotation data and signature data does not match.")
    }
  }
  else if (nrow(annotationData) == ncol(inputData)) {
    if (!all(rownames(annotationData) == colnames(inputData))) {
      stop("Annotation data and signature data does not match.")
    }
    inputData <- t(inputData)
  }
  else {
    stop("Annotation data and signature data does not match.")
  }
  pathwaydata <- t(inputData)
  if (scale) {
    pathwaydata <- t(scale(t(pathwaydata)))
  }
  boxplotdf <- data.frame(t(pathwaydata), Group = annotationData[, 
                                                                 1])
  boxplotdfm <- reshape2::melt(boxplotdf, value.name = "Score", 
                               variable.name = "Signature", id.vars = "Group")
  theplot <- ggplot2::ggplot(boxplotdfm, ggplot2::aes(Group, 
                                                      Score)) + ggplot2::facet_wrap(~Signature, scales = "free", 
                                                                                    nrow = nrow, ncol = ncol)
  if (violinPlot) {
    theplot <- theplot + ggplot2::geom_violin(ggplot2::aes(fill = Group)) + 
      ggplot2::theme_classic()
  }
  else {
    theplot <- theplot + ggplot2::geom_boxplot(outlier.shape = NA, 
                                               ggplot2::aes(fill = Group), notch = notch) + 
      ggplot2::theme_classic()
  }
  if (includePoints) {
    theplot <- theplot + ggplot2::geom_point(position = ggplot2::position_jitter(width = 0.1))
  }
  if (rotateLabels) {
    theplot <- theplot + ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, 
                                                                            hjust = 1))
  }
  if (is.null(fill_colors)) {
    if (n < 3) 
      n <- 3
    fill_colors <- RColorBrewer::brewer.pal(n, "Set1")
  }
  return(theplot + ggplot2::scale_fill_manual(values = fill_colors) + 
           ggplot2::ggtitle(name))
}

```






### TB risk signatures for parasites Visit_1 {.tabset}
```{r, messAGE=TRUE}
indata <- indata_v1
indata <- se
TBsignatures$'Leong_RISK_29_up' = c("SPSB1","MOB3C","FUT4","DERA","WARS","ZNRF1","SLC3A1","SRBD1",
                                    "HM13", "CTSA","TCN2", "APOL6","NAGA","IL31RA")
TBsignatures$'Leong_RISK_29_dn' = c("SH2D1B","AGAP9","ZNF202","NEIL1","LUC7L","CCDC78","PRSS53","ENO3",
                                    "CIRBP","SNRNP70","ZNF419","CCDC14","GSTA4","MDN1","SPDYE5")

TBsignatures$Suliman_RISK_4 <- c("GAS6",  "SEPTIN4", "CD1C",  "BLK" )
TBsignatures$Zak_RISK_16 <- c("ANKRD22",  "APOL1",    "BATF2" ,   "ETV7",     "FCGR1A",   "FCGR1B" ,  "GBP1" ,      "GBP2" , "GBP4", "GBP5", "SCARF1",   "SEPTIN4",    "SERPING1", "STAT1" ,   "TAP1", "TRAFD1")

# select signatures to test TB progression risk
samp_tbsigs <- list(TBsignatures$Suliman_RISK_4, TBsignatures$Zak_RISK_16, TBsignatures$Sweeney_OD_3, TBsignatures$Darboe_RISK_11, TBsignatures$Leong_RISK_29, TBsignatures$Leong_RISK_29_up, TBsignatures$Leong_RISK_29_dn)

names(samp_tbsigs) <- c("Suliman_RISK_4", "Zak_RISK_16", "Sweeney_OD_3", "Darboe_RISK_11", "Leong_RISK_29", "Leong_RISK_29_up","Leong_RISK_29_dn")

# samp_tbsigs <- list(TBsignatures$Suliman_RISK_4)
# 
# names(samp_tbsigs) <- c("Suliman_RISK_4")


gsva_res <- runTBsigProfiler(indata, useAssay = "log_counts_cpm", signatures=samp_tbsigs, algorithm="GSVA")

ssgsea_res <- runTBsigProfiler(indata, useAssay = "log_counts_cpm", signatures=samp_tbsigs, algorithm="ssGSEA")

plAGE_res <- runTBsigProfiler(indata, useAssay = "log_counts_cpm", signatures=samp_tbsigs, algorithm="PLAGE")

```



#### ssGSEA {.tabset}
```{r }
signatureHeatmap(ssgsea_res, name="ssGSEA", signatureColNames = names(samp_tbsigs),
                 annotationColNames = c("HELMINTH"),
                 showColumnNames = FALSE, scale = TRUE,
                 split_heatmap='none')
```


#### Art's fix
```{r}
AsignatureBoxplot=function (inputData, annotationData, signatureColNames, annotationColName, 
  name = "Signatures", scale = FALSE, violinPlot = FALSE, 
  includePoints = TRUE, notch = FALSE, rotateLabels = FALSE, 
  nrow = NULL, ncol = NULL, fill_colors = NULL) 
{
  if (methods::is(inputData, "SummarizedExperiment")) {
    if (any(duplicated(signatureColNames))) {
      stop("Duplicate signature column name is not supported.")
    }
    if (!all(signatureColNames %in% colnames(SummarizedExperiment::colData(inputData)))) {
      stop("Signature column name not found in inputData.")
    }
    if (!all(annotationColName %in% colnames(SummarizedExperiment::colData(inputData)))) {
      stop("Annotation column name not found in inputData.")
    }
    annotationData <- data.frame(SummarizedExperiment::colData(inputData)[, 
      annotationColName, drop = FALSE])
    inputData <- data.frame(SummarizedExperiment::colData(inputData)[, 
      signatureColNames, drop = FALSE])
  }
  else {
    if (ncol(annotationData) != 1) {
      stop("annotationData must have only one column.")
    }
    annotationColName <- colnames(annotationData)
  }
  if (length(annotationColName) != 1) {
    stop("You must specify a single annotation column name to color boxplots by.")
  }
  if (!is.factor(annotationData[, 1])) {
    annotationData[, 1] <- as.factor(annotationData[, 1])
  }
  n <- length(levels(annotationData[, 1]))
  if (n > 9) {
    stop("Too many levels in the annotation data. The boxplot can contain a maximum of 9 levels")
  }
  if (nrow(annotationData) == nrow(inputData)) {
    if (!all(rownames(annotationData) == rownames(inputData))) {
      stop("Annotation data and signature data does not match.")
    }
  }
  else if (nrow(annotationData) == ncol(inputData)) {
    if (!all(rownames(annotationData) == colnames(inputData))) {
      stop("Annotation data and signature data does not match.")
    }
    inputData <- t(inputData)
  }
  else {
    stop("Annotation data and signature data does not match.")
  }
  pathwaydata <- t(inputData)
  if (scale) {
    pathwaydata <- t(scale(t(pathwaydata)))
  }
  boxplotdf <- data.frame(t(pathwaydata), Group = annotationData[, 
    1])
  boxplotdfm <- reshape2::melt(boxplotdf, value.name = "Score", 
    variable.name = "Signature", id.vars = "Group")
  theplot <- ggplot2::ggplot(boxplotdfm, ggplot2::aes(Group, 
    Score)) + ggplot2::facet_wrap(~Signature, scales = "free", 
    nrow = nrow, ncol = ncol)
  if (violinPlot) {
    theplot <- theplot + ggplot2::geom_violin(ggplot2::aes(fill = Group)) + 
      ggplot2::theme_classic()
  }
  else {
    theplot <- theplot + ggplot2::geom_boxplot(outlier.shape = NA, 
      ggplot2::aes(fill = Group), notch = notch) + 
      ggplot2::theme_classic()
  }
  if (includePoints) {
    theplot <- theplot + ggplot2::geom_point(position = ggplot2::position_jitter(width = 0.1))
  }
  if (rotateLabels) {
    theplot <- theplot + ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 45, 
      hjust = 1))
  }
  if (is.null(fill_colors)) {
    if (n < 3) 
      n <- 3
    fill_colors <- RColorBrewer::brewer.pal(n, "Set1")
  }
  return(theplot + ggplot2::scale_fill_manual(values = fill_colors) + 
    ggplot2::ggtitle(name))
}
```

#### Boxplot {.tabset}
```{r}
# colData(ssgsea_res)[,"status"] <- as.factor(colData(ssgsea_res)[,"status"])
samp_tbsigs$Zak_RISK_16 <- NULL
samp_tbsigs$Sweeney_OD_3 <- NULL
colData(ssgsea_res)$nutrition <- gsub("mal", "Mal", colData(ssgsea_res)$nutrition)
colData(ssgsea_res)$nutrition <- gsub("well", "Well", colData(ssgsea_res)$nutrition)
colData(ssgsea_res)$nutrition_visit <- paste(colData(ssgsea_res)$nutrition, colData(ssgsea_res)$visit, sep="_")
colData(ssgsea_res)$group <- paste(colData(ssgsea_res)$nutrition, colData(ssgsea_res)$HELMINTH, colData(ssgsea_res)$visit, sep="_")




```
### dots and lines from visit 1 to visit 6 {.tabset}
```{r}
# dataframe of colData
df.col <- as.data.frame(colData(ssgsea_res))
# subset visit 1 and visit 6
df.col <- filter(df.col, visit=="Visit_1" | visit=="Visit_6")

# subset only individuals in both timepoints
x <- filter(df.col, visit=="Visit_1")
y <- filter(df.col, visit=="Visit_6")

all(y$TB.LION.ID.Number. %in% x$TB.LION.ID.Number.)
###FALSE  

x <-  filter(x, x$ID1 %in% y$ID1)
df.col.1 <- filter(df.col, ID1 %in% x$ID1)
# graph BMI, helminth status, risk signatures ssGSEA scores/PLAGE scores
df.col.1 %>% group_by(visit) %>% mutate(ind = 1:length(visit)) -> df.col.1

```

# BMI {.tabset}
```{r}
#dir.create("results/prelim")
# ggplot(data=df.col.1, mapping=aes(visit, BMI, group=ind, linetype= ID1)) + 
#   geom_point(aes(color=nutrition, shape=HELMINTH)) +
#   geom_line()
# ggsave("results/prelim/bmi_change.png")
# ggsave("results/prelim/bmi_change.svg")
```


# risk signature score change ssgsea {.tabset}
```{r}

p_RISK11 <- ggplot(data=df.col.1, mapping=aes(visit, Darboe_RISK_11, group=ID1, linetype= ID1)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()+ theme(legend.position = "none")

p_PRED29 <- ggplot(data=df.col.1, mapping=aes(visit, Leong_RISK_29, group=ind, linetype= ID1)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()

  

p_RISK11 + theme(legend.position = "none")+ 
p_PRED29+ theme(legend.position = "none") 

#ggsave("results/prelim/ssgsea_change.png")
#ggsave("results/prelim/ssgsea_change.svg")

sigboxes <- AsignatureBoxplot(ssgsea_res, name="ssGSEA", signatureColNames= c("Darboe_RISK_11", "Leong_RISK_29"),
                 annotationColName = c("nutrition_visit"), rotateLabels = T,scale = TRUE) + geom_segment(inherit.aes = T, mapping=aes(visit, Darboe_RISK_11, group=ID1, linetype= ID1)
sigboxes 
```


```{r}
ggsave(file.path("outs/TBSigProfiler", "nutriton_visit.svg"), width=4, height=4)
ggsave(file.path("outs/TBSigProfiler", "nutriton_visit.png"), width=7, height=7)
```




```{r}
# df <- as.data.frame(colData(ssgsea_res))
# 
# group <- df$status
# score <- df$Suliman_RISK_4
# 
# 
# groups <- list(df$Suliman_RISK_4, df$Leong_RISK_29, df$Zak_RISK_16, df$Sweeney_OD_3)
# names(groups) <- c("Suliman_RISK4", "Predict29", "Risk16", "Sweeney3")
# 
# 
# sigs <- c("Suliman_RISK_4", "Leong_RISK_29", "Zak_RISK_16", "Sweeney_OD_3")
# 
# 
# 
# annotationData <- data.frame(SummarizedExperiment::colData(inputData)[, 
#       annotationColName, drop = FALSE])
# n <- length(levels(annotationData[, 1]))
# inputData <- data.frame(SummarizedExperiment::colData(inputData)[, 
#       signatureColNames, drop = FALSE])
# 
# ### transpose input data or no?
# pathwaydata <- t(inputData)
# pathwaydata <- t(scale(t(pathwaydata)))
# 
# boxplotdf <- data.frame(t(pathwaydata), Group = annotationData[, 
#     1])
# 
# boxplotdfm <- reshape2::melt(boxplotdf, value.name = "Score", 
#     variable.name = "Signature", id.vars = "Group")
# 
# 
#   boxplotdfm$Group <- as.factor(boxplotdfm$Group)
#   fill_colors <- RColorBrewer::brewer.pal(4, "Set1")
#   
#   
# theplot <- ggplot(boxplotdfm, aes(x=Group, y=Score, fill=Group)) +                   
#       geom_boxplot() + 
#       theme_classic()+ 
#       scale_fill_manual(values = fill_colors) + 
#       ggtitle(name) +
#       facet_wrap(~Signature, scales = "free")
# theplot


```

```{r }
#svg("results/status_boxplots_ssgsea.svg", height=4, width = 3.5, pointsize = 8)
# AsignatureBoxplot(ssgsea_res, name="ssGSEA", signatureColNames = names(samp_tbsigs),
#                  annotationColName = c("status"), rotateLabels = T,scale = TRUE) +
#    theme(axis.text.x = ggplot2::element_text(angle = 45,
#        hjust = 1, size=8)) +
#     theme(legend.text = element_text(size = 8))
# 
# 
# 
# ggsave("results/status_boxplots_ssgsea_visit_1.png", height=7, width = 7,  pointsize = 5, units="in", device = "png")
# 
# ggsave("results/status_boxplots_ssgsea_visit_1.svg", height=7, width = 7,  pointsize = 5, units="in", device = "svg")
```




#### Boxplots Single {.tabset}

```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureBoxplot(ssgsea_res, name=i, signatureColNames = i,
                 annotationColName = c("HELMINTH"), rotateLabels = T))

  cat("\n\n")
}

```

#### Signature plots {.tabset}
```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  signatureGeneHeatmap(ssgsea_res, useAssay="log_counts_cpm",
                     samp_tbsigs[[i]],
                     name = i, signatureColNames = NULL,
                     annotationColNames = c("HELMINTH",i),
                     showColumnNames = FALSE, 
                     column_order =  NULL)

  cat("\n\n")
}

```


#### AUC Table {.tabset}
```{r }
set.seed(0)
tableAUC(ssgsea_res,
         annotationColName = "HELMINTH",
         signatureColNames = names(samp_tbsigs),
         num.boot = 100,
         pb.show = FALSE)
```

#### AUC Boxplots {.tabset}
```{r }
#svg(filename = "t.svg", pointsize = 8, width = 6.5, height = 3.8) 
set.seed(0)
compareBoxplots(ssgsea_res, annotationColName = "HELMINTH",
                signatureColNames = names(samp_tbsigs),
                pb.show = FALSE, fill.col = "blue",
                rotateLabels = TRUE)
#dev.off()
```

#### AUC Boxplots, selected sigs {.tabset}
```{r }

out <- tableAUC(ssgsea_res,
         annotationColName = "HELMINTH",
         signatureColNames = names(samp_tbsigs),
         num.boot = 100,
         pb.show = FALSE, 
         output="data.frame")
# Use dplyr::filter to only select rows with AUC > 0.6 on out object 
# Pipe into a dplyr::select function to only select signames %>% unlist %>% unname
# Save this as signatureColNames

out <- out %>% 
          dplyr::filter(P.value<0.05) %>% 
          dplyr::select(Signature) %>% 
          unlist() %>% 
          unname()


#png(filename = "RePORT-ssgsea-auc-boxplot-bl.png", pointsize = 8, width = 6.5, height = 3.8)
# set.seed(0)
# compareBoxplots(ssgsea_res, annotationColName = "HELMINTH",
#                 signatureColNames = out,
#                 pb.show = FALSE, fill.col = "blue",
#                 rotateLabels = TRUE)
#dev.off()
```


#### ROC plots {.tabset}
```{r , fig.height = 9, fig.width = 12}
signatureROCplot_CI(inputData = ssgsea_res,
                   signatureColNames = names(samp_tbsigs),
                   annotationColName = "HELMINTH")

```

#### Separate ROC plots  {.tabset}

```{r , results = 'asis'}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureROCplot_CI(inputData = ssgsea_res,
                   signatureColNames = i,
                   annotationColName = "HELMINTH",
                   name = paste("ROC plot,", i, sep = " ")))

  cat("\n\n")
}
```


### PLAGE {.tabset}


#### Heatmap {.tabset}

```{r }
signatureHeatmap(plAGE_res, name="PLAGE", signatureColNames = names(samp_tbsigs),
                 annotationColNames = c("HELMINTH"),
                 showColumnNames = FALSE,
                 split_heatmap='none')
```


#### Boxplot {.tabset}

```{r }
AsignatureBoxplot(plAGE_res, name="PLAGE", signatureColNames = names(samp_tbsigs),
                 annotationColName = c("HELMINTH"))# , rotateLabels = TRUE)
```


#### Boxplot plAGE separated mal and helminth {.tabset}
```{r }
#svg("results/status_boxplots_ssgsea.svg", height=4, width = 3.5, pointsize = 8)
AsignatureBoxplot(plAGE_res, name="PLAGE", signatureColNames = names(samp_tbsigs),
                 annotationColName = c("status"), rotateLabels = T,scale = TRUE) +
  theme(axis.text.x = ggplot2::element_text(angle = 45, 
      hjust = 1, size=8)) +
   theme(legend.text = element_text(size = 8))

AsignatureBoxplot(ssgsea_res, name="ssGSEA Visit 1", signatureColNames = names(samp_tbsigs),
                 annotationColName = c("status"), rotateLabels = T,scale = TRUE) +
  theme(axis.text.x = ggplot2::element_text(angle = 45, 
      hjust = 1, size=8)) +
   theme(legend.text = element_text(size = 8))

ggsave("outs/TBSigProfiler/boxplots/status_boxplots_ssGSEA_visit_1.png", height=7, width = 7,  pointsize = 5, units="in", device = "png")
# 
# ggsave("results/status_boxplots_plAGE_visit_1.svg", height=7, width = 7,  pointsize = 5, units="in", device = "svg")
```
#### Boxplots Single {.tabset}

```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureBoxplot(plAGE_res,
                         name=i, 
                         signatureColNames = i,
                         annotationColName = c("HELMINTH"),   
                         ##violinPlot = TRUE,
                         rotateLabels = T))

  cat("\n\n")
}

```

#### Signature plots {.tabset}
```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  signatureGeneHeatmap(plAGE_res, useAssay="log_counts_cpm",
                     samp_tbsigs[[i]],
                     name = i, signatureColNames = NULL,
                     annotationColNames = c("HELMINTH",i),
                     showColumnNames = FALSE, 
                     column_order = NULL)

  cat("\n\n")
}

```

#### AUC Table {.tabset}
```{r }
set.seed(0)
tableAUC(plAGE_res,
         annotationColName = "HELMINTH",
         signatureColNames = names(samp_tbsigs),
         num.boot = 100,
         pb.show = FALSE)
```

#### AUC Boxplots {.tabset}
```{r }
set.seed(0)
compareBoxplots(plAGE_res, annotationColName = "HELMINTH",
                signatureColNames = names(samp_tbsigs),
                pb.show = FALSE, fill.col = "blue",
                rotateLabels = TRUE)
```


```{r }
set.seed(0)
compareBoxplots(plAGE_res, annotationColName = "nutrition",
                signatureColNames = names(samp_tbsigs),
                pb.show = FALSE, fill.col = "blue",
                rotateLabels = TRUE)
```

#### ROC plots {.tabset}
```{r , fig.height = 9, fig.width = 12}
signatureROCplot_CI(inputData = plAGE_res,
                   signatureColNames = names(samp_tbsigs),
                   annotationColName = "HELMINTH")

```

#### Separate ROC plots  {.tabset}

```{r , results = 'asis'}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureROCplot_CI(inputData = plAGE_res,
                   signatureColNames = i,
                   annotationColName = "HELMINTH",
                   name = paste("ROC plot,", i, sep = " ")))

  cat("\n\n")
}
```




### TB risk signatures for malnourished {.tabset}

### ssGSEA {.tabset}
```{r }
signatureHeatmap(ssgsea_res, name="ssGSEA", signatureColNames = names(samp_tbsigs),
                 annotationColNames = c("nutrition"),
                 showColumnNames = FALSE, scale = TRUE,
                 split_heatmap='none')
```

#### Boxplot {.tabset}

```{r }
signatureBoxplot(ssgsea_res, name="ssGSEA", signatureColNames = names(samp_tbsigs),
                 annotationColName = c("nutrition"), scale = TRUE) #rotateLabels = TRUE,
```



#### Boxplots Single {.tabset}

```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureBoxplot(ssgsea_res, name=i, signatureColNames = i,
                 annotationColName = c("nutrition"), rotateLabels = T))

  cat("\n\n")
}

```

#### Signature plots {.tabset}
```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  signatureGeneHeatmap(ssgsea_res, useAssay="log_counts_cpm",
                     samp_tbsigs[[i]],
                     name = i, signatureColNames = NULL,
                     annotationColNames = c("nutrition",i),
                     showColumnNames = FALSE, 
                     column_order =  NULL)

  cat("\n\n")
}

```


#### AUC Table {.tabset}
```{r }
set.seed(0)
tableAUC(ssgsea_res,
         annotationColName = "nutrition",
         signatureColNames = names(samp_tbsigs),
         num.boot = 100,
         pb.show = FALSE)
```

#### AUC Boxplots {.tabset}
```{r }
#svg(filename = "t.svg", pointsize = 8, width = 6.5, height = 3.8) 
set.seed(0)
compareBoxplots(ssgsea_res, annotationColName = "nutrition",
                signatureColNames = names(samp_tbsigs),
                pb.show = FALSE, fill.col = "blue",
                rotateLabels = TRUE)
#dev.off()
```

#### AUC Boxplots, selected sigs {.tabset}
```{r }

out <- tableAUC(ssgsea_res,
         annotationColName = "nutrition",
         signatureColNames = names(samp_tbsigs),
         num.boot = 100,
         pb.show = FALSE, 
         output="data.frame")
# Use dplyr::filter to only select rows with AUC > 0.6 on out object 
# Pipe into a dplyr::select function to only select signames %>% unlist %>% unname
# Save this as signatureColNames

out <- out %>% 
         dplyr::filter(P.value<0.05) %>% 
        dplyr::select(Signature) %>% 
          unlist() %>% 
          unname()


#png(filename = "RePORT-ssgsea-auc-boxplot-bl.png", pointsize = 8, width = 6.5, height = 3.8)
# set.seed(0)
# compareBoxplots(ssgsea_res, annotationColName = "nutrition",
#                 signatureColNames = out,
#                 pb.show = FALSE, fill.col = "blue",
#                 rotateLabels = TRUE)
# dev.off()
```


#### ROC plots {.tabset}
```{r , fig.height = 9, fig.width = 12}
signatureROCplot_CI(inputData = ssgsea_res,
                   signatureColNames = names(samp_tbsigs),
                   annotationColName = "nutrition")

```

#### Separate ROC plots  {.tabset}

```{r , results = 'asis'}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureROCplot_CI(inputData = ssgsea_res,
                   signatureColNames = i,
                   annotationColName = "nutrition",
                   name = paste("ROC plot,", i, sep = " ")))

  cat("\n\n")
}
```


### PLAGE {.tabset}


#### Heatmap {.tabset}

```{r }
signatureHeatmap(plAGE_res, name="PLAGE", signatureColNames = names(samp_tbsigs),
                 annotationColNames = c("nutrition"),
                 showColumnNames = FALSE,
                 split_heatmap='none')
```


#### Boxplot {.tabset}

```{r }
signatureBoxplot(plAGE_res, name="PLAGE", signatureColNames = names(samp_tbsigs),
                 annotationColName = c("nutrition"))# , rotateLabels = TRUE)
```

#### Boxplots Single {.tabset}

```{r, results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureBoxplot(plAGE_res,
                         name=i, 
                         signatureColNames = i,
                         annotationColName = c("nutrition"),   
                         ##violinPlot = TRUE,
                         rotateLabels = T))

  cat("\n\n")
}

```

#### Signature plots {.tabset}
```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  signatureGeneHeatmap(plAGE_res, useAssay="log_counts_cpm",
                     samp_tbsigs[[i]],
                     name = i, signatureColNames = NULL,
                     annotationColNames = c("nutrition",i),
                     showColumnNames = FALSE, 
                     column_order = NULL)

  cat("\n\n")
}

```

#### AUC Table {.tabset}
```{r }
set.seed(0)
tableAUC(plAGE_res,
         annotationColName = "nutrition",
         signatureColNames = names(samp_tbsigs),
         num.boot = 100,
         pb.show = FALSE)
```

#### AUC Boxplots {.tabset}
```{r }
set.seed(0)
compareBoxplots(plAGE_res, annotationColName = "nutrition",
                signatureColNames = names(samp_tbsigs),
                pb.show = FALSE, fill.col = "blue",
                rotateLabels = TRUE)
```

#### ROC plots {.tabset}
```{r , fig.height = 9, fig.width = 12}
signatureROCplot_CI(inputData = plAGE_res,
                   signatureColNames = names(samp_tbsigs),
                   annotationColName = "nutrition")

```

#### Separate ROC plots  {.tabset}

```{r , results = 'asis'}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureROCplot_CI(inputData = plAGE_res,
                   signatureColNames = i,
                   annotationColName = "nutrition",
                   name = paste("ROC plot,", i, sep = " ")))

  cat("\n\n")
}
```

### TB Signature Profiler for Visit_6 {.tabset}
```{r, messAGE=TRUE}

indata <- indata_v6

gsva_res <- runTBsigProfiler(indata, useAssay = "log_counts_cpm", signatures=samp_tbsigs, algorithm="GSVA")

ssgsea_res <- runTBsigProfiler(indata, useAssay = "log_counts_cpm", signatures=samp_tbsigs, algorithm="ssGSEA")

plAGE_res <- runTBsigProfiler(indata, useAssay = "log_counts_cpm", signatures=samp_tbsigs, algorithm="PLAGE")
```


#### ssGSEA {.tabset}
```{r }
signatureHeatmap(ssgsea_res, name="ssGSEA", signatureColNames = names(samp_tbsigs),
                 annotationColNames = c("HELMINTH"),
                 showColumnNames = FALSE, scale = TRUE,
                 split_heatmap='none')
```

#### Boxplot {.tabset}

```{r }
signatureBoxplot(ssgsea_res, name="ssGSEA", signatureColNames = names(samp_tbsigs),
                 annotationColName = c("HELMINTH"), scale = TRUE) #rotateLabels = TRUE,
```
```{r}
AsignatureBoxplot(ssgsea_res, name="ssGSEA Visit 6", signatureColNames = names(samp_tbsigs),
                 annotationColName = c("status"), rotateLabels = T,scale = TRUE) +
  theme(axis.text.x = ggplot2::element_text(angle = 45, 
      hjust = 1, size=8)) +
   theme(legend.text = element_text(size = 8))

ggsave("outs/TBSigProfiler/boxplots/status_boxplots_ssGSEA_visit_6.png", height=7, width = 7,  pointsize = 5, units="in", device = "png")
```



#### Boxplots Single {.tabset}

```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureBoxplot(ssgsea_res, name=i, signatureColNames = i,
                 annotationColName = c("HELMINTH"), rotateLabels = T))

  cat("\n\n")
}

```

#### Signature plots {.tabset}
```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  signatureGeneHeatmap(ssgsea_res, useAssay="log_counts_cpm",
                     samp_tbsigs[[i]],
                     name = i, signatureColNames = NULL,
                     annotationColNames = c("HELMINTH",i),
                     showColumnNames = FALSE, 
                     column_order =  NULL)

  cat("\n\n")
}

```


#### AUC Table {.tabset}
```{r }
set.seed(0)
tableAUC(ssgsea_res,
         annotationColName = "HELMINTH",
         signatureColNames = names(samp_tbsigs),
         num.boot = 100,
         pb.show = FALSE)
```

#### AUC Boxplots {.tabset}
```{r}
#svg(filename = "t.svg", pointsize = 8, width = 6.5, height = 3.8) 
set.seed(0)
compareBoxplots(ssgsea_res, annotationColName = "HELMINTH",
                signatureColNames = names(samp_tbsigs),
                pb.show = FALSE, fill.col = "blue",
                rotateLabels = TRUE)
#dev.off()
```

#### AUC Boxplots, selected sigs {.tabset}
```{r }

out <- tableAUC(ssgsea_res,
         annotationColName = "HELMINTH",
         signatureColNames = names(samp_tbsigs),
         num.boot = 100,
         pb.show = FALSE, 
         output="data.frame")
# Use dplyr::filter to only select rows with AUC > 0.6 on out object 
# Pipe into a dplyr::select function to only select signames %>% unlist %>% unname
# Save this as signatureColNames

out <- out %>% 
         dplyr::filter(P.value<0.05) %>% 
        dplyr::select(Signature) %>% 
          unlist() %>% 
          unname()


# png(filename = "RePORT-ssgsea-auc-boxplot-bl.png", pointsize = 8, width = 6.5, height = 3.8)
# set.seed(0)
# compareBoxplots(ssgsea_res, annotationColName = "HELMINTH",
#                 signatureColNames = out,
#                 pb.show = FALSE, fill.col = "blue",
#                 rotateLabels = TRUE)
# dev.off()
```


#### ROC plots {.tabset}
```{r , fig.height = 9, fig.width = 12}
signatureROCplot_CI(inputData = ssgsea_res,
                   signatureColNames = names(samp_tbsigs),
                   annotationColName = "HELMINTH")

```

#### Separate ROC plots  {.tabset}

```{r , results = 'asis'}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureROCplot_CI(inputData = ssgsea_res,
                   signatureColNames = i,
                   annotationColName = "HELMINTH",
                   name = paste("ROC plot,", i, sep = " ")))

  cat("\n\n")
}
```


### PLAGE {.tabset}


#### Heatmap {.tabset}

```{r }
signatureHeatmap(plAGE_res, name="PLAGE", signatureColNames = names(samp_tbsigs),
                 annotationColNames = c("HELMINTH"),
                 showColumnNames = FALSE,
                 split_heatmap='none')
```


#### Boxplot {.tabset}

```{r }
signatureBoxplot(plAGE_res, name="PLAGE", signatureColNames = names(samp_tbsigs),
                 annotationColName = c("HELMINTH"))# , rotateLabels = TRUE)
```

#### Boxplots Single {.tabset}

```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureBoxplot(plAGE_res,
                         name=i, 
                         signatureColNames = i,
                         annotationColName = c("HELMINTH"),   
                         ##violinPlot = TRUE,
                         rotateLabels = T))

  cat("\n\n")
}

```

#### Signature plots {.tabset}
```{r , results="asis"}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  signatureGeneHeatmap(plAGE_res, useAssay="log_counts_cpm",
                     samp_tbsigs[[i]],
                     name = i, signatureColNames = NULL,
                     annotationColNames = c("HELMINTH",i),
                     showColumnNames = FALSE, 
                     column_order = NULL)

  cat("\n\n")
}

```

#### AUC Table {.tabset}
```{r }
set.seed(0)
tableAUC(plAGE_res,
         annotationColName = "HELMINTH",
         signatureColNames = names(samp_tbsigs),
         num.boot = 100,
         pb.show = FALSE)
```

#### AUC Boxplots {.tabset}
```{r }
set.seed(0)
compareBoxplots(plAGE_res, annotationColName = "HELMINTH",
                signatureColNames = names(samp_tbsigs),
                pb.show = FALSE, fill.col = "blue",
                rotateLabels = TRUE)
```

#### ROC plots {.tabset}
```{r , fig.height = 9, fig.width = 12}
signatureROCplot_CI(inputData = plAGE_res,
                   signatureColNames = names(samp_tbsigs),
                   annotationColName = "HELMINTH")

```

#### Separate ROC plots  {.tabset}

```{r , results = 'asis'}
for (i in names(samp_tbsigs)){

  cat("#####", i, "\n")

  print(signatureROCplot_CI(inputData = plAGE_res,
                   signatureColNames = i,
                   annotationColName = "HELMINTH",
                   name = paste("ROC plot,", i, sep = " ")))

  cat("\n\n")
}
```




### TB risk signatures for malnourished {.tabset}

<!-- ## ssGSEA {.tabset} -->
<!-- ```{r } -->
<!-- signatureHeatmap(ssgsea_res, name="ssGSEA", signatureColNames = names(samp_tbsigs), -->
<!--                  annotationColNames = c("nutrition"), -->
<!--                  showColumnNames = FALSE, scale = TRUE, -->
<!--                  split_heatmap='none') -->
<!-- ``` -->

<!-- #### Boxplot {.tabset} -->

<!-- ```{r } -->
<!-- signatureBoxplot(ssgsea_res, name="ssGSEA", signatureColNames = names(samp_tbsigs), -->
<!--                  annotationColName = c("nutrition"), scale = TRUE) #rotateLabels = TRUE, -->
<!-- ``` -->



<!-- #### Boxplots Single {.tabset} -->

<!-- ```{r , results="asis"} -->
<!-- for (i in names(samp_tbsigs)){ -->

<!--   cat("#####", i, "\n") -->

<!--   print(signatureBoxplot(ssgsea_res, name=i, signatureColNames = i, -->
<!--                  annotationColName = c("nutrition"), rotateLabels = T)) -->

<!--   cat("\n\n") -->
<!-- } -->

<!-- ``` -->

<!-- #### Signature plots {.tabset} -->
<!-- ```{r , results="asis"} -->
<!-- for (i in names(samp_tbsigs)){ -->

<!--   cat("#####", i, "\n") -->

<!--   signatureGeneHeatmap(ssgsea_res, useAssay="log_counts_cpm", -->
<!--                      samp_tbsigs[[i]], -->
<!--                      name = i, signatureColNames = NULL, -->
<!--                      annotationColNames = c("nutrition",i), -->
<!--                      showColumnNames = FALSE,  -->
<!--                      column_order =  NULL) -->

<!--   cat("\n\n") -->
<!-- } -->

<!-- ``` -->


<!-- #### AUC Table {.tabset} -->
<!-- ```{r } -->
<!-- set.seed(0) -->
<!-- tableAUC(ssgsea_res, -->
<!--          annotationColName = "nutrition", -->
<!--          signatureColNames = names(samp_tbsigs), -->
<!--          num.boot = 100, -->
<!--          pb.show = FALSE) -->
<!-- ``` -->

<!-- #### AUC Boxplots {.tabset} -->
<!-- ```{r } -->
<!-- #svg(filename = "t.svg", pointsize = 8, width = 6.5, height = 3.8)  -->
<!-- set.seed(0) -->
<!-- compareBoxplots(ssgsea_res, annotationColName = "nutrition", -->
<!--                 signatureColNames = names(samp_tbsigs), -->
<!--                 pb.show = FALSE, fill.col = "blue", -->
<!--                 rotateLabels = TRUE) -->
<!-- #dev.off() -->
<!-- ``` -->

<!-- #### AUC Boxplots, selected sigs {.tabset} -->
<!-- ```{r } -->

<!-- out <- tableAUC(ssgsea_res, -->
<!--          annotationColName = "nutrition", -->
<!--          signatureColNames = names(samp_tbsigs), -->
<!--          num.boot = 100, -->
<!--          pb.show = FALSE,  -->
<!--          output="data.frame") -->
<!-- # Use dplyr::filter to only select rows with AUC > 0.6 on out object  -->
<!-- # Pipe into a dplyr::select function to only select signames %>% unlist %>% unname -->
<!-- # Save this as signatureColNames -->

<!-- out <- out %>%  -->
<!--          dplyr::filter(P.value<0.05) %>%  -->
<!--         dplyr::select(Signature) %>%  -->
<!--           unlist() %>%  -->
<!--           unname() -->


<!-- #png(filename = "RePORT-ssgsea-auc-boxplot-bl.png", pointsize = 8, width = 6.5, height = 3.8) -->
<!-- # set.seed(0) -->
<!-- # compareBoxplots(ssgsea_res, annotationColName = "nutrition", -->
<!-- #                 signatureColNames = out, -->
<!-- #                 pb.show = FALSE, fill.col = "blue", -->
<!-- #                 rotateLabels = TRUE) -->
<!-- # dev.off() -->
<!-- ``` -->


<!-- #### ROC plots {.tabset} -->
<!-- ```{r , fig.height = 9, fig.width = 12} -->
<!-- signatureROCplot_CI(inputData = ssgsea_res, -->
<!--                    signatureColNames = names(samp_tbsigs), -->
<!--                    annotationColName = "nutrition") -->

<!-- ``` -->

<!-- #### Separate ROC plots  {.tabset} -->

<!-- ```{r , results = 'asis'} -->
<!-- for (i in names(samp_tbsigs)){ -->

<!--   cat("#####", i, "\n") -->

<!--   print(signatureROCplot_CI(inputData = ssgsea_res, -->
<!--                    signatureColNames = i, -->
<!--                    annotationColName = "nutrition", -->
<!--                    name = paste("ROC plot,", i, sep = " "))) -->

<!--   cat("\n\n") -->
<!-- } -->
<!-- ``` -->


<!-- ## PLAGE {.tabset} -->


<!-- #### Heatmap -->

<!-- ```{r } -->
<!-- signatureHeatmap(plAGE_res, name="PLAGE", signatureColNames = names(samp_tbsigs), -->
<!--                  annotationColNames = c("nutrition"), -->
<!--                  showColumnNames = FALSE, -->
<!--                  split_heatmap='none') -->
<!-- ``` -->


<!-- #### Boxplot {.tabset} -->

<!-- ```{r } -->
<!-- signatureBoxplot(plAGE_res, name="PLAGE", signatureColNames = names(samp_tbsigs), -->
<!--                  annotationColName = c("nutrition"))# , rotateLabels = TRUE) -->
<!-- ``` -->

<!-- #### Boxplots Single {.tabset} -->

<!-- ```{r , results="asis"} -->
<!-- for (i in names(samp_tbsigs)){ -->

<!--   cat("#####", i, "\n") -->

<!--   print(signatureBoxplot(plAGE_res, -->
<!--                          name=i,  -->
<!--                          signatureColNames = i, -->
<!--                          annotationColName = c("nutrition"),    -->
<!--                          ##violinPlot = TRUE, -->
<!--                          rotateLabels = T)) -->

<!--   cat("\n\n") -->
<!-- } -->

<!-- ``` -->

<!-- #### Signature plots {.tabset} -->
<!-- ```{r , results="asis"} -->
<!-- for (i in names(samp_tbsigs)){ -->

<!--   cat("#####", i, "\n") -->

<!--   signatureGeneHeatmap(plAGE_res, useAssay="log_counts_cpm", -->
<!--                      samp_tbsigs[[i]], -->
<!--                      name = i, signatureColNames = NULL, -->
<!--                      annotationColNames = c("nutrition",i), -->
<!--                      showColumnNames = FALSE,  -->
<!--                      column_order = NULL) -->

<!--   cat("\n\n") -->
<!-- } -->

<!-- ``` -->

<!-- #### AUC Table {.tabset} -->
<!-- ```{r } -->
<!-- set.seed(0) -->
<!-- tableAUC(plAGE_res, -->
<!--          annotationColName = "nutrition", -->
<!--          signatureColNames = names(samp_tbsigs), -->
<!--          num.boot = 100, -->
<!--          pb.show = FALSE) -->
<!-- ``` -->

<!-- #### AUC Boxplots {.tabset} -->
<!-- ```{r } -->
<!-- set.seed(0) -->
<!-- compareBoxplots(plAGE_res, annotationColName = "nutrition", -->
<!--                 signatureColNames = names(samp_tbsigs), -->
<!--                 pb.show = FALSE, fill.col = "blue", -->
<!--                 rotateLabels = TRUE) -->
<!-- ``` -->

<!-- #### ROC plots {.tabset} -->
<!-- ```{r , fig.height = 9, fig.width = 12} -->
<!-- signatureROCplot_CI(inputData = plAGE_res, -->
<!--                    signatureColNames = names(samp_tbsigs), -->
<!--                    annotationColName = "nutrition") -->

<!-- ``` -->

<!-- #### Separate ROC plots  {.tabset} -->

<!-- ```{r, results = 'asis'} -->
<!-- for (i in names(samp_tbsigs)){ -->

<!--   cat("#####", i, "\n") -->

<!--   print(signatureROCplot_CI(inputData = plAGE_res, -->
<!--                    signatureColNames = i, -->
<!--                    annotationColName = "nutrition", -->
<!--                    name = paste("ROC plot,", i, sep = " "))) -->

<!--   cat("\n\n") -->
<!-- } -->
<!-- ``` -->


# Longitudinal visualization {.tabset}
```{r, messAGE=TRUE}
indata <- se
indata <- mkAssay(indata, log = T)


# select signatures to test TB progression risk
samp_tbsigs <- list(TBsignatures$Suliman_RISK_4, TBsignatures$Zak_RISK_16, TBsignatures$Sweeney_OD_3, TBsignatures$Darboe_RISK_11, TBsignatures$Leong_RISK_29)

names(samp_tbsigs) <- c("Suliman_RISK_4", "Zak_RISK_16", "Sweeney_OD_3", "Darboe_RISK_11", "Leong_RISK_29")


gsva_res <- runTBsigProfiler(indata, useAssay = "log_counts_cpm", signatures=samp_tbsigs, algorithm="GSVA")

ssgsea_res <- runTBsigProfiler(indata, useAssay = "log_counts_cpm", signatures=samp_tbsigs, algorithm="ssGSEA")

plAGE_res <- runTBsigProfiler(indata, useAssay = "log_counts_cpm", signatures=samp_tbsigs, algorithm="PLAGE")
```

## ssGSEA {.tabset}
### dots and lines from visit 1 to visit 6 {.tabset}
```{r}
# dataframe of colData
df.col <- as.data.frame(colData(ssgsea_res))
# subset visit 1 and visit 6
df.col <- dplyr::filter(df.col, visit=="Visit_1" | visit=="Visit_6")

# subset only individuals in both timepoints
x <- dplyr::filter(df.col, visit=="Visit_1")
y <- dplyr::filter(df.col, visit=="Visit_6")

all(y$ID1 %in% x$ID1)
###FALSE  

x <-  dplyr::filter(x, x$ID1 %in% y$ID1) # list of matching V1 V2
x.2 <-  dplyr::filter(x, !(x$ID1 %in% y$ID1)) # list of non-matching V1 V2


df.col.1 <- dplyr::filter(df.col, ID1 %in% x$ID1)
# graph BMI, helminth status, risk signatures ssGSEA scores/PLAGE scores
df.col.1 %>% group_by(visit) %>% mutate(ind = 1:length(visit)) -> df.col.1

```

# BMI {.tabset}
```{r}
# Need updated V2 BMI data
#dir.create("results/prelim")
ggplot(data=df.col.1, mapping=aes(visit, BMI, group=ind, linetype= ID1)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()
# ggsave("results/prelim/bmi_change.png")
# ggsave("results/prelim/bmi_change.svg")
```


# risk signature score change ssgsea {.tabset}
```{r}

p_RISK4 <- ggplot(data=df.col.1, mapping=aes(visit, Suliman_RISK_4, group=ind, linetype= ID1)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()

p_RISK11 <- ggplot(data=df.col.1, mapping=aes(visit, Darboe_RISK_11, group=ind, linetype= ID1)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()

p_PRED29 <- ggplot(data=df.col.1, mapping=aes(visit, Leong_RISK_29, group=ind, linetype= ID1)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()

p_SWEENEY3 <- ggplot(data=df.col.1, mapping=aes(visit, Sweeney_OD_3, group=ind, linetype= ID1)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()
  
p_RISK4 + theme(legend.position = "none")+
p_RISK11 + theme(legend.position = "none")+ 
p_PRED29+ theme(legend.position = "none") +
p_SWEENEY3 + guides(ID1="none")

# ggsave("results/prelim/ssgsea_change.png")
# ggsave("results/prelim/ssgsea_change.svg")


```


# PlAGE change {.tabset}
```{r}
# dataframe of colData
pf.col <- as.data.frame(colData(plAGE_res))
# subset visit 1 and visit 6
pf.col <- dplyr::filter(pf.col, visit=="Visit_1" | visit=="Visit_6")

# subset only individuals in both timepoints
x <- dplyr::filter(pf.col, visit=="Visit_1")
y <- dplyr::filter(pf.col, visit=="Visit_6")

all(y$TB.LION.ID.Number. %in% x$TB.LION.ID.Number.)
###FALSE  

x <-  dplyr::filter(x, x$TB.LION.ID.Number. %in% y$TB.LION.ID.Number.)
pf.col.1 <- dplyr::filter(pf.col, TB.LION.ID.Number. %in% x$TB.LION.ID.Number.)
# graph BMI, helminth status, risk signatures ssGSEA scores/PLAGE scores
pf.col.1 %>% group_by(visit) %>% mutate(ind = 1:length(visit)) -> pf.col.1

```

```{r}
p_RISK4 <- ggplot(data=pf.col.1, mapping=aes(visit, Suliman_RISK_4, group=ind, linetype= TB.LION.ID.Number.)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()

p_RISK11 <- ggplot(data=pf.col.1, mapping=aes(visit, Darboe_RISK_11, group=ind, linetype= TB.LION.ID.Number.)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()

p_PRED29 <- ggplot(data=pf.col.1, mapping=aes(visit, Leong_RISK_29, group=ind, linetype= TB.LION.ID.Number.)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()

p_SWEENEY3 <- ggplot(data=pf.col.1, mapping=aes(visit, Sweeney_OD_3, group=ind, linetype= TB.LION.ID.Number.)) + 
  geom_point(aes(color=nutrition, shape=HELMINTH)) +
  geom_line()
  
p_RISK4 +
p_RISK11 + 
p_PRED29 +
p_SWEENEY3 
```

# animalcules {.tabset}
## metadata for animalcules {.tabset}
```{r}
text <- read_tsv(file.path("data/microbiome_subjectlist_file.txt"))
subjdata <- read_excel("data/microbiome_subjdata_complete.xlsx")
text <- text[ which(grepl("_R1.fastq.gz",text$file.txt)),]
text <- gsub("_R1.fastq.gz", "", text$file.txt)
text <- as.data.frame(text)


# edit data
df <- subjdata %>% mutate(TB.LION.ID.Number. = gsub("_Visit.*", "", text)) %>% 
  mutate(visit=str_extract(text, 'Visit_.')) %>% 
  mutate(nutrition=ifelse(BMI>=18.5,"well", "mal")) %>% 
  mutate(status=paste(HELMINTH, nutrition, sep="_")) %>% 
  rename(c('TB.LION.ID.Number.'='subjid')) 
  

x <- gsub("_Visit.*", "", text$text)
#x <- unique(x)
y <- gsub("_Visit.*", "", df$subjid)
y <- unique(y)

all(x %in% y)
# TRUE

df <- df[,-1]
df[64:65,"Gender"] <- "Female"

#write.csv(df, "microbiome_metadata.csv")
```



## animalcules results from shiny app {.tabset}
```{r}

#genus <- read.delim(pipe("pbpaste"))
#write.csv(x, "data/animalcules_diffex_deseq.csv")

x <- read.csv(file.path("data/animalcules_diffex_deseq.csv"), header=T, )

x$X <- NULL

Absent_well <- dplyr::filter(x, grepl("Absent_well", x$control)==T)
Present_well <- dplyr::filter(x, grepl("Present_well", x$control)==T)
helminthspecific_well <- Present_well[which(grepl("Absent_well", Present_well$experiment)),]
helminthspecific_mal <- x[which(grepl("Absent_mal vs. Present_mal", x$Contrast)),]
nut_spec_helminthpresent <- x[which(grepl("Present_mal vs. Present_well", x$Contrast)),]
nut_spec_helminthabsent <- x[which(grepl("Absent_mal vs. Absent_well", x$Contrast)),]



```

## Genus level {.tabset}
```{r}
#genus <- read.delim(pipe("pbpaste"))
#write.csv(genus, "data/animalcules_diffex_deseq_genus.csv")

genus <- read.csv("data/animalcules_diffex_deseq_genus.csv")

ghelminthspecific_well <- genus[which(grepl("Absent_well vs. Present_well", genus$Contrast)),]
ghelminthspecific_mal <- genus[which(grepl("Absent_mal vs. Present_mal", genus$Contrast)),]
gnut_spec_helminthpresent <- genus[which(grepl("Present_mal vs. Present_well", genus$Contrast)),]
gnut_spec_helminthabsent <- genus[which(grepl("Absent_mal vs. Absent_well", genus$Contrast)),]


# get specific genera and spp stats
topgen <- c("Megasphaera", "Veillonella", "Lacrimispora", "Turicibacter")
topspp <- c("Turicibacter sanguinis", "Butyrivibrio proteoclasticus", "Bifidobacterium gallicum", "Megasphaera elsdenii")

gendf <- genus %>% dplyr::filter(microbe %in% topgen)
sppdf <- x %>% dplyr::filter(microbe %in% topspp)

```

# Microbiome plots {.tabset}

![Figure for microbiome data at visit 1](results/animalcules_plots/All_plots_V2.png)



# Table 1 for visit 1 RNAseq {.tabset}
```{r}
# read data
df.2 <- read.csv("data/cleaned_subjdata_tblion_rnaseq.csv")

# filter for visit 1
df.2 <- df.2 %>% dplyr::filter(visit=="Visit_1")
table(df.2$HELMINTH, df.2$nutrition)

```
```{r}
table(v1$HELMINTH, v1$nutrition)
```

# Table 1 for microbiome {.tabset}
```{r}
mae <- read.csv("data/subjdata_tblion_microbiome.csv")
```
```{r}
mv1 <- dplyr::filter(mae, visit=="Visit_2")
table(mv1$HELMINTH, mv1$nutrition)
```

# Nutrient Prediction
```{r}
#devtools::install_github('KChen-lab/METAFlux')
# tutorial
# https://htmlpreview.github.io/?https://github.com/KChen-lab/METAFlux/blob/main/Tutorials/pipeline.html

# citation
# Huang, Y., Mohanty, V., Dede, M. et al. Characterizing cancer metabolism from bulk and single-cell RNA-seq data using METAFlux. Nat Commun 14, 4883 (2023). https://doi.org/10.1038/s41467-023-40457-w


library(METAFlux)
#load the toy example
data("bulk_test_example")
# medium file for human derived samples
data("human_blood")
#Calculate mras for human sample data
scores<-calculate_reaction_score(bulk_test_example)


######### Start here
mat = as.matrix(assay(se,"log_counts"))

scores<-calculate_reaction_score(mat)


#Calculate flux for human sample data
flux<-compute_flux(mras=scores,medium=human_blood) 

#optional: flux scores cubic root normalization
cbrt <- function(x) {
    sign(x) * abs(x)^(1/3)
}

flux=cbrt(flux)

```


```{r}

data("nutrient_lookup_files")
glucose<-data.frame(glucose=flux[grep("HMR_9034",human_gem$ID),])
glucose 
```

```{r}
# Join boxplots of glucose with colData
#df <- as.data.frame(colData(se))

#df.1 <- merge(df, glucose, by.x)
```

```{r}
ggplot(glucose,aes(y=-glucose,x="sample"))+geom_boxplot()+ggtitle("Glucose uptake level")+xlab("")+ylab("Glucose uptake scores")
```


```{r}
#compute pathway level activity for all samples
pathway<-unique(unlist(human_gem$SUBSYSTEM))
pathway_score<-list()
for (i in pathway){
path=i
activity_score<-c()
for (d in 1:ncol(flux)){
activity_score[d]<-mean(abs(flux[which(unlist(human_gem$SUBSYSTEM)==i),d]))
} 
pathway_score[[i]]<-activity_score
}

all_pathway_score<-as.data.frame(do.call(rbind,pathway_score))
colnames(all_pathway_score) <- colnames(flux)

#heatmap 
mapal <- colorRampPalette(RColorBrewer::brewer.pal(11,"RdBu"))(256)

# annotation
df=data.frame(Helminth=as.factor(colData(se)$"HELMINTH"),
              Nutrition=as.factor(colData(se)$"nutrition"),
              Visit=as.factor(colData(se)$"visit"),
              ID=colData(se)$"subjid")
df <- column_to_rownames(df, var="ID")

df.1 <- df %>% arrange(Nutrition, Helminth, Visit)

mat <- all_pathway_score[,row.names(df.1)]

ha = HeatmapAnnotation(df = df, col = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")))

png(filename = file.path("outs/metabolism/metab_heatmap.png"), width=2100, height = 2100)
pheatmap::pheatmap(mat,
                   annotation_col = df,
                   annotation_colors = list(Nutrition=c("mal"="Red","well"="Blue"), Helminths=c("Present"="Black", "Absent"="Yellow")),
                   show_colnames = FALSE,
                   cluster_cols =F,
                   color = rev(mapal),
                   scale = "row")
dev.off()
```
Need to use ANOVA/t.test to get p-values for enrichment scores of each metabolic pathway

## Redo
```{r}
# these 3 did not make it through rsubread
## mismatched fastq lines and other error
#2, 24, 72;"LION_032500502LB_VSIT_6"  "LION_033300511LB_VISIT_6" "LION_044800738LB_VISIT_6"
```


# Session Info
```{r, results='markup', messAGE=TRUE}
sessionInfo()
```

